{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import pickle\n",
    "import ujson\n",
    "import sys\n",
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from tqdm import tqdm\n",
    "from collections import defaultdict\n",
    "\n",
    "from bigbio.dataloader import BigBioConfigHelpers\n",
    "from bigbio.utils.constants import Tasks\n",
    "\n",
    "\n",
    "sys.path.append('..')\n",
    "from bigbio_utils import CUIS_TO_REMAP, CUIS_TO_EXCLUDE, DATASET_NAMES\n",
    "from bigbio_utils import dataset_to_documents, dataset_to_df, resolve_abbreviation\n",
    "\n",
    "conhelps = BigBioConfigHelpers()\n",
    "\n",
    "pd.set_option('display.max_rows', 200)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_overlap(df, dataset_name=None, plot=False):\n",
    "    \"\"\"\n",
    "    Overlap between train mentions and test mentions.\n",
    "\n",
    "    Return both the % of entities that overlap and the % of mentions with eneities that appear in train set (since some entities appear more than once)\n",
    "    \"\"\"\n",
    "    # Get unique entities in data splits\n",
    "    train_df = df.query('split == \"train\"')\n",
    "    test_df = df.query('split == \"test\"')\n",
    "    train_valid_df = df.query('split != \"test\"')\n",
    "    unique_train = set(train_df.db_id.unique())\n",
    "    unique_test = set(test_df.db_id.unique())\n",
    "    unique_train_valid = set(train_valid_df.db_id.unique())\n",
    "    unique_ents = df.db_id.unique()\n",
    "\n",
    "    # Get total number of unique entities by type\n",
    "    ents_by_type = df.groupby(\"type\").agg({\"db_id\": \"nunique\"}).to_dict()['db_id']\n",
    "    mention_counts_by_type = df.groupby(\"type\").agg({\"db_id\": \"count\"}).to_dict()['db_id']\n",
    "\n",
    "    # Get distribution of entities\n",
    "    if plot:\n",
    "        test_df[\"in_train\"] = test_df.db_id.isin(unique_train)\n",
    "        curie_counts = (\n",
    "            test_df.groupby(\"db_id\")\n",
    "            .agg({\"mention_id\": \"count\", \"in_train\": \"first\"})\n",
    "            .rename({\"mention_id\": \"cui_num_test_mentions\"}, axis=1)\n",
    "        )\n",
    "        sns.displot(\n",
    "            data=curie_counts, x=\"cui_num_test_mentions\", hue=\"in_train\", log_scale=True\n",
    "        )\n",
    "        if dataset_name is not None:\n",
    "            plt.title(dataset_name)\n",
    "        plt.show()\n",
    "\n",
    "    # Get overlap of mentions\n",
    "    train_test_ent_overlap = len(unique_test.intersection(unique_train)) / len(\n",
    "        unique_test\n",
    "    )\n",
    "    test_ent_overlap = len(unique_test.intersection(unique_train_valid)) / len(\n",
    "        unique_test\n",
    "    )\n",
    "    mention_overlap = test_df[\"db_id\"].isin(unique_train).mean()\n",
    "    return {\n",
    "        \"unique_ents\": len(unique_ents),\n",
    "        \"ent_overlap\": train_test_ent_overlap,\n",
    "        \"mention_overlap\": mention_overlap,\n",
    "        # \"ent_counts_by_type\": ents_by_type,\n",
    "        # 'mention_counts_by_type': mention_counts_by_type,\n",
    "        'unique_types': len(df.type.unique()),\n",
    "        \"total_documents\": len(df.pmid.unique()),\n",
    "        \"train_documents\": len(train_df.pmid.unique()),\n",
    "        \"test_documents\": len(test_df.pmid.unique()),\n",
    "        \"total_mentions\": df.shape[0],\n",
    "        \"train_mentions\": train_df.shape[0],\n",
    "        \"test_mentions\": test_df.shape[0],\n",
    "        \"has_validation_set\": (\"validation\" in df.split.unique()),\n",
    "    }\n",
    "\n",
    "\n",
    "def dataset_to_df(dataset):\n",
    "    '''\n",
    "    Convert BigBio dataset to pandas DataFrame\n",
    "    '''\n",
    "    columns = [\n",
    "        \"pmid\",\n",
    "        \"mention_id\",\n",
    "        \"text\",\n",
    "        \"type\",\n",
    "        \"db_name\",\n",
    "        \"db_id\",\n",
    "        \"split\",\n",
    "    ]\n",
    "    all_lines = []\n",
    "    for split in dataset.keys():\n",
    "        for doc in dataset[split]:\n",
    "            pmid = doc[\"document_id\"]\n",
    "            for e in doc[\"entities\"]:\n",
    "                if len(e[\"normalized\"]) == 0:\n",
    "                    continue\n",
    "                text = \" \".join(e[\"text\"])\n",
    "                db_name = e[\"normalized\"][0][\"db_name\"]\n",
    "                db_id = e[\"normalized\"][0][\"db_id\"]\n",
    "                all_lines.append(\n",
    "                    [pmid, e[\"id\"], text, e[\"type\"], db_name, db_id, split]\n",
    "                )\n",
    "\n",
    "    return pd.DataFrame(all_lines, columns=columns)\n",
    "\n",
    "\n",
    "\n",
    "# Basic tests to see if functions work\n",
    "# bc5cdr_df = dataset_to_df(bc5cdr_bigbio)\n",
    "# get_overlap(bc5cdr_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/7 [00:00<?, ?it/s]Reusing dataset med_mentions_dataset (/Users/david/.cache/huggingface/datasets/med_mentions_dataset/medmentions_full_bigbio_kb/1.0.0/b5c8691186d4701f9b18eddbe36d178ccf7e55761dcc6140c57f4410754511ac)\n",
      "100%|██████████| 3/3 [00:00<00:00, 389.67it/s]\n",
      " 14%|█▍        | 1/7 [00:06<00:40,  6.77s/it]Reusing dataset med_mentions_dataset (/Users/david/.cache/huggingface/datasets/med_mentions_dataset/medmentions_st21pv_bigbio_kb/1.0.0/b5c8691186d4701f9b18eddbe36d178ccf7e55761dcc6140c57f4410754511ac)\n",
      "100%|██████████| 3/3 [00:00<00:00, 286.54it/s]\n",
      " 29%|██▊       | 2/7 [00:10<00:25,  5.07s/it]Reusing dataset bc5cdr_dataset (/Users/david/.cache/huggingface/datasets/bc5cdr_dataset/bc5cdr_bigbio_kb/1.0.0/f01f16ea9b65ead985bedadf7335195c32297c8f1b09417fc607b102a6757d6f)\n",
      "100%|██████████| 3/3 [00:00<00:00, 394.37it/s]\n",
      " 43%|████▎     | 3/7 [00:11<00:12,  3.20s/it]Reusing dataset gnormplus_dataset (/Users/david/.cache/huggingface/datasets/gnormplus_dataset/gnormplus_bigbio_kb/1.0.0/1bb16f1b4abf9394b9180cac70edc575cc5e7d32c697f8b9f69ba2f643d2fc95)\n",
      "100%|██████████| 2/2 [00:00<00:00, 448.32it/s]\n",
      " 57%|█████▋    | 4/7 [00:11<00:06,  2.01s/it]Reusing dataset ncbi_disease_dataset (/Users/david/.cache/huggingface/datasets/ncbi_disease_dataset/ncbi_disease_bigbio_kb/1.0.0/10a393201e55b403e5d107701b719368f54f1bf3d3438a1233f99be0badeb034)\n",
      "100%|██████████| 3/3 [00:00<00:00, 510.38it/s]\n",
      " 71%|███████▏  | 5/7 [00:11<00:02,  1.35s/it]Reusing dataset nlm_chem_dataset (/Users/david/.cache/huggingface/datasets/nlm_chem_dataset/nlmchem_bigbio_kb/1.0.0/d91131823c66b7dd1162027991ea47c342e478209b37cf261c5f122d30409594)\n",
      "100%|██████████| 3/3 [00:00<00:00, 495.35it/s]\n",
      " 86%|████████▌ | 6/7 [00:13<00:01,  1.25s/it]Reusing dataset nlm_gene_dataset (/Users/david/.cache/huggingface/datasets/nlm_gene_dataset/nlm_gene_bigbio_kb/1.0.0/4291402c7589961b34294745522bffe7e289d8f6c11d99a66d93362c0d30187d)\n",
      "100%|██████████| 2/2 [00:00<00:00, 384.46it/s]\n",
      "100%|██████████| 7/7 [00:13<00:00,  1.91s/it]\n"
     ]
    }
   ],
   "source": [
    "all_stats = []\n",
    "for dataset in tqdm(['medmentions_full','medmentions_st21pv','bc5cdr','gnormplus','ncbi_disease','nlmchem', 'nlm_gene']):\n",
    "    data = conhelps.for_config_name(f\"{dataset}_bigbio_kb\").load_dataset()\n",
    "    df = dataset_to_df(data)\n",
    "    stats = get_overlap(df, dataset)\n",
    "    stats['dataset'] = dataset\n",
    "    all_stats.append(stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'unique_ents': 34724,\n",
       "  'ent_overlap': 0.6199371930107094,\n",
       "  'mention_overlap': 0.8221204512041217,\n",
       "  'unique_types': 127,\n",
       "  'total_documents': 4392,\n",
       "  'train_documents': 2635,\n",
       "  'test_documents': 879,\n",
       "  'total_mentions': 385098,\n",
       "  'train_mentions': 230591,\n",
       "  'test_mentions': 76861,\n",
       "  'has_validation_set': True,\n",
       "  'dataset': 'medmentions_full'},\n",
       " {'unique_ents': 25419,\n",
       "  'ent_overlap': 0.5754995861416577,\n",
       "  'mention_overlap': 0.7740867096645666,\n",
       "  'unique_types': 21,\n",
       "  'total_documents': 4392,\n",
       "  'train_documents': 2635,\n",
       "  'test_documents': 879,\n",
       "  'total_mentions': 203282,\n",
       "  'train_mentions': 122241,\n",
       "  'test_mentions': 40157,\n",
       "  'has_validation_set': True,\n",
       "  'dataset': 'medmentions_st21pv'},\n",
       " {'unique_ents': 2348,\n",
       "  'ent_overlap': 0.5300380228136882,\n",
       "  'mention_overlap': 0.7733048693707432,\n",
       "  'unique_types': 2,\n",
       "  'total_documents': 1500,\n",
       "  'train_documents': 500,\n",
       "  'test_documents': 500,\n",
       "  'total_mentions': 29044,\n",
       "  'train_mentions': 9494,\n",
       "  'test_mentions': 9837,\n",
       "  'has_validation_set': True,\n",
       "  'dataset': 'bc5cdr'},\n",
       " {'unique_ents': 1353,\n",
       "  'ent_overlap': 0.07894736842105263,\n",
       "  'mention_overlap': 0.08377288240769469,\n",
       "  'unique_types': 2,\n",
       "  'total_documents': 533,\n",
       "  'train_documents': 279,\n",
       "  'test_documents': 254,\n",
       "  'total_mentions': 6252,\n",
       "  'train_mentions': 3029,\n",
       "  'test_mentions': 3223,\n",
       "  'has_validation_set': False,\n",
       "  'dataset': 'gnormplus'},\n",
       " {'unique_ents': 789,\n",
       "  'ent_overlap': 0.6699507389162561,\n",
       "  'mention_overlap': 0.815625,\n",
       "  'unique_types': 4,\n",
       "  'total_documents': 792,\n",
       "  'train_documents': 592,\n",
       "  'test_documents': 100,\n",
       "  'total_mentions': 6881,\n",
       "  'train_mentions': 5134,\n",
       "  'test_mentions': 960,\n",
       "  'has_validation_set': True,\n",
       "  'dataset': 'ncbi_disease'},\n",
       " {'unique_ents': 1787,\n",
       "  'ent_overlap': 0.4747225647348952,\n",
       "  'mention_overlap': 0.6228987993138937,\n",
       "  'unique_types': 1,\n",
       "  'total_documents': 150,\n",
       "  'train_documents': 80,\n",
       "  'test_documents': 50,\n",
       "  'total_mentions': 37999,\n",
       "  'train_mentions': 21044,\n",
       "  'test_mentions': 11660,\n",
       "  'has_validation_set': True,\n",
       "  'dataset': 'nlmchem'},\n",
       " {'unique_ents': 3348,\n",
       "  'ent_overlap': 0.48191365227537925,\n",
       "  'mention_overlap': 0.5408348457350273,\n",
       "  'unique_types': 5,\n",
       "  'total_documents': 550,\n",
       "  'train_documents': 450,\n",
       "  'test_documents': 100,\n",
       "  'total_mentions': 15553,\n",
       "  'train_mentions': 12798,\n",
       "  'test_mentions': 2755,\n",
       "  'has_validation_set': False,\n",
       "  'dataset': 'nlm_gene'}]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "| Dataset                                                                                                            | Total Documents   | Total Mentions   | Unique Entities   |   Unique Types |   Entity Overlap |   Mention Overlap | Source Documents   | Linked Ontology   |\n",
      "|:-------------------------------------------------------------------------------------------------------------------|:------------------|:-----------------|:------------------|---------------:|-----------------:|------------------:|:-------------------|:------------------|\n",
      "| MedMentions Full [(Mohan and Li, 2019)](https://github.com/chanzuckerberg/MedMentions)                             | 4,392             | 385,098          | 34,724            |            127 |           0.6199 |            0.8221 | PubMed Abstracts   | UMLS              |\n",
      "| MedMentions ST21PV [(Mohan and Li, 2019)](https://github.com/chanzuckerberg/MedMentions)                           | 4,392             | 203,282          | 25,419            |             21 |           0.5755 |            0.7741 | PubMed Abstracts   | UMLS              |\n",
      "| BC5CDR [(Li et al, 2016)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4860626)                                    | 1,500             | 29,044           | 2,348             |              2 |           0.53   |            0.7733 | PubMed Abstracts   | MeSH              |\n",
      "| GNormPlus [(Wei et al, 2016)](https://www.ncbi.nlm.nih.gov/research/bionlp/Tools/gnormplus)                        | 533               | 6,252            | 1,353             |              2 |           0.0789 |            0.0838 | PubMed Abstracts   | Entrez            |\n",
      "| NCBI Disease [(Dogan et al, 2014)](https://www.sciencedirect.com/science/article/pii/S1532046413001974?via%3Dihub) | 792               | 6,881            | 789               |              4 |           0.67   |            0.8156 | PubMed Abstracts   | MeSH, OMIM        |\n",
      "| NLM Chem [(Islamaj et al, 2021)](https://www.nature.com/articles/s41597-021-00875-1)                               | 150               | 37,999           | 1,787             |              1 |           0.4747 |            0.6229 | PMC Full-Text      | MeSH              |\n",
      "| NLM Gene [(Islamaj et al, 2021)](https://doi.org/10.1016/j.jbi.2021.103779)                                        | 550               | 15,553           | 3,348             |              5 |           0.4819 |            0.5408 | PMC Full-Text      | Entrez            |\n",
      "\\begin{tabular}{llllrll}\n",
      "\\toprule\n",
      "  \\textbf{Dataset} & \\textbf{Num Docs} & \\textbf{Mentions} & \\textbf{Unique Ents} &  \\textbf{Ent Types} & \\textbf{Doc Type} & \\textbf{Ontology} \\\\\n",
      "\\midrule\n",
      "  MedMentions Full &             4,392 &           385,098 &               34,724 &                 127 &  PubMed Abstracts &              UMLS \\\\\n",
      "MedMentions ST21PV &             4,392 &           203,282 &               25,419 &                  21 &  PubMed Abstracts &              UMLS \\\\\n",
      "            BC5CDR &             1,500 &            29,044 &                2,348 &                   2 &  PubMed Abstracts &              MeSH \\\\\n",
      "         GNormPlus &               533 &             6,252 &                1,353 &                   2 &  PubMed Abstracts &            Entrez \\\\\n",
      "      NCBI Disease &               792 &             6,881 &                  789 &                   4 &  PubMed Abstracts &        MeSH, OMIM \\\\\n",
      "          NLM Chem &               150 &            37,999 &                1,787 &                   1 &     PMC Full-Text &              MeSH \\\\\n",
      "          NLM Gene &               550 &            15,553 &                3,348 &                   5 &     PMC Full-Text &            Entrez \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\n",
      "\\begin{tabular}{lrr}\n",
      "\\toprule\n",
      "  \\textbf{Dataset} &  \\textbf{Ent. Overlap} &  \\textbf{Ment. Overlap} \\\\\n",
      "\\midrule\n",
      "  MedMentions Full &                 0.6199 &                  0.8221 \\\\\n",
      "MedMentions ST21PV &                 0.5755 &                  0.7741 \\\\\n",
      "            BC5CDR &                 0.5300 &                  0.7733 \\\\\n",
      "         GNormPlus &                 0.0789 &                  0.0838 \\\\\n",
      "      NCBI Disease &                 0.6700 &                  0.8156 \\\\\n",
      "          NLM Chem &                 0.4747 &                  0.6229 \\\\\n",
      "          NLM Gene &                 0.4819 &                  0.5408 \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dataset_to_doc_type = {\n",
    "    \"medmentions_full\": \"PubMed Abstracts\",\n",
    "    \"medmentions_st21pv\": \"PubMed Abstracts\",\n",
    "    \"bc5cdr\": \"PubMed Abstracts\",\n",
    "    \"gnormplus\": \"PubMed Abstracts\",\n",
    "    \"ncbi_disease\": \"PubMed Abstracts\",\n",
    "    \"nlmchem\": \"PMC Full-Text\",\n",
    "    \"craft\": \"PMC Full-Text\",\n",
    "    \"bc6id\": \"PMC Figure Captions\",\n",
    "    \"bc3gm\": \"PMC Full-Text\",\n",
    "    \"plantnorm\": \"PubMed Abstracts\",\n",
    "    \"nlm_gene\": \"PMC Full-Text\",\n",
    "}\n",
    "\n",
    "dataset_to_ontology = {\n",
    "    \"medmentions_full\": \"UMLS\",\n",
    "    \"medmentions_st21pv\": \"UMLS\",\n",
    "    \"bc5cdr\": \"MeSH\",\n",
    "    \"gnormplus\": \"Entrez\",\n",
    "    \"ncbi_disease\": \"MeSH, OMIM\",\n",
    "    \"nlmchem\": \"MeSH\",\n",
    "    \"craft\": \"Many\",\n",
    "    \"bc6id\": \"Many\",\n",
    "    \"bc3gm\": \"Entrez\",\n",
    "    \"plantnorm\": \"NCBI Taxonomy\",\n",
    "    'nlm_gene': 'Entrez',\n",
    "}\n",
    "\n",
    "dataset_to_pretty_name = {\n",
    "    \"medmentions_full\": \"MedMentions Full\",\n",
    "    \"medmentions_st21pv\": \"MedMentions ST21PV\",\n",
    "    \"bc5cdr\": \"BC5CDR\",\n",
    "    \"gnormplus\": \"GNormPlus\",\n",
    "    \"ncbi_disease\": \"NCBI Disease\",\n",
    "    \"nlmchem\": \"NLM Chem\",\n",
    "    \"craft\": \"CRAFT\",\n",
    "    \"bc6id\": \"BC6ID\",\n",
    "    \"bc3gm\": \"BC3GM\",\n",
    "    \"plantnorm\": \"PlantNorm\",\n",
    "    'nlm_gene': \"NLM Gene\"\n",
    "}\n",
    "\n",
    "dataset_to_paper_url = {\n",
    "    \"medmentions_full\": \"[(Mohan and Li, 2019)](https://github.com/chanzuckerberg/MedMentions)\",\n",
    "    \"medmentions_st21pv\": \"[(Mohan and Li, 2019)](https://github.com/chanzuckerberg/MedMentions)\",\n",
    "    \"bc5cdr\": \"[(Li et al, 2016)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4860626)\",\n",
    "    \"gnormplus\": \"[(Wei et al, 2016)](https://www.ncbi.nlm.nih.gov/research/bionlp/Tools/gnormplus)\",\n",
    "    \"ncbi_disease\": \"[(Dogan et al, 2014)](https://www.sciencedirect.com/science/article/pii/S1532046413001974?via%3Dihub)\",\n",
    "    \"nlmchem\": \"[(Islamaj et al, 2021)](https://www.nature.com/articles/s41597-021-00875-1)\",\n",
    "    \"craft\": \"[(Bada et al, 2012)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3476437)\",\n",
    "    \"bc6id\": \"[(Arighi et al, 2017)](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-S8-S2)\",\n",
    "    \"bc3gm\": \"[(Lu et al, 2011)](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-S8-S2)\",\n",
    "    \"plantnorm\": \"[(Cho et al, 2017)](https://pubmed.ncbi.nlm.nih.gov/29029598)\",\n",
    "    'nlm_gene': \"[(Islamaj et al, 2021)](https://doi.org/10.1016/j.jbi.2021.103779)\"\n",
    "}\n",
    "\n",
    "dataset_to_citation = {\n",
    "    \"medmentions_full\": \"mohan2019medmentions\",\n",
    "    \"medmentions_st21pv\": \"mohan2019medmentions\",\n",
    "    \"bc5cdr\": \"li2016bc5cdr\",\n",
    "    \"gnormplus\": \"wei2015gnormplus\",\n",
    "    \"ncbi_disease\": \"dougan2014ncbi\",\n",
    "    \"nlmchem\": \"islamaj2021nlm_chem\",\n",
    "    \"craft\": \"bada2012craft_concept\",\n",
    "    \"bc6id\": \"arighi2017bc6id\",\n",
    "    \"bc3gm\": \"lu2011bc3gn\",\n",
    "    \"plantnorm\": \"cho2017plantnorm\",\n",
    "    'nlm_gene': \"islamaj2021nlm_gene\"\n",
    "}\n",
    "\n",
    "output = pd.DataFrame.from_records(all_stats)\n",
    "output[\"source_documents\"] = output.dataset.map(dataset_to_doc_type)\n",
    "output[\"linked_ontology\"] = output.dataset.map(dataset_to_ontology)\n",
    "for col in [\"unique_ents\", \"total_documents\", \"total_mentions\"]:\n",
    "    output[col] = output[col].map(lambda x: \"{:,}\".format(x))\n",
    "\n",
    "output_subset = (\n",
    "    output[\n",
    "        [\n",
    "            \"dataset\",\n",
    "            \"total_documents\",\n",
    "            \"total_mentions\",\n",
    "            \"unique_ents\",\n",
    "            'unique_types',\n",
    "            \"ent_overlap\",\n",
    "            \"mention_overlap\",\n",
    "            \"source_documents\",\n",
    "            \"linked_ontology\",\n",
    "        ]\n",
    "    ]\n",
    "    .rename({\"unique_ents\": \"unique_entities\", \"ent_overlap\": \"entity_overlap\"}, axis=1)\n",
    "    .round(4)\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "# Make Markdown Tables\n",
    "markdown_output = output_subset.copy()\n",
    "markdown_output[\"dataset\"] = markdown_output.dataset.map(\n",
    "    lambda x: dataset_to_pretty_name[x] + \" \" + dataset_to_paper_url[x]\n",
    ")\n",
    "markdown_output.columns = [\n",
    "    \" \".join(x.split(\"_\")).title() for x in markdown_output.columns\n",
    "]\n",
    "print(markdown_output.to_markdown(index=False))\n",
    "\n",
    "\n",
    "# Make Latex Tables\n",
    "latex_output = output_subset.copy()\n",
    "latex_output[\"dataset\"] = latex_output.dataset.map(lambda x: dataset_to_pretty_name[x])\n",
    "# latex_output['dataset'] = latex_output.dataset.map(lambda x: dataset_to_pretty_name[x] + ' ' + '\\cite{%s}' % dataset_to_citation[x])\n",
    "# latex_output.columns = [\" \".join(x.split(\"_\")).title() for x in latex_output.columns]\n",
    "\n",
    "latex_output = latex_output.rename(\n",
    "    {\n",
    "        \"dataset\": \"Dataset\",\n",
    "        \"total_documents\": \"Num Docs\",\n",
    "        \"total_mentions\": \"Mentions\",\n",
    "        \"unique_entities\": \"Unique Ents\",\n",
    "        'unique_types': 'Ent Types',\n",
    "        \"entity_overlap\": \"Ent. Overlap\",\n",
    "        \"mention_overlap\": \"Ment. Overlap\",\n",
    "        \"source_documents\": \"Doc Type\",\n",
    "        \"linked_ontology\": \"Ontology\",\n",
    "    },\n",
    "    axis=1,\n",
    ")\n",
    "latex_output_1 = latex_output[[\"Dataset\", \"Num Docs\", \"Mentions\",\"Unique Ents\",'Ent Types', 'Doc Type', 'Ontology']]\n",
    "latex_output_2 = latex_output[[\"Dataset\", 'Ent. Overlap', \"Ment. Overlap\"]]\n",
    "\n",
    "latex_output_1.columns = [\"\\textbf{\" + x + '}' for x in latex_output_1.columns]\n",
    "latex_output_2.columns = [\"\\textbf{\" + x + '}' for x in latex_output_2.columns]\n",
    "\n",
    "print(latex_output_1.to_latex(index=False, escape=False, bold_rows=True))\n",
    "print(latex_output_2.to_latex(index=False, escape=False, bold_rows=True))\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;31mSignature:\u001b[0m\n",
      "\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_latex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mbuf\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mcol_space\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mna_rep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'NaN'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mformatters\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mfloat_format\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0msparsify\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mindex_names\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mbold_rows\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mcolumn_format\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mlongtable\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mescape\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mdecimal\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'.'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mmulticolumn\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mmulticolumn_format\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mmultirow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mcaption\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mlabel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m    \u001b[0mposition\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\n",
      "\u001b[0;34m\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mDocstring:\u001b[0m\n",
      "Render object to a LaTeX tabular, longtable, or nested table.\n",
      "\n",
      "Requires ``\\usepackage{booktabs}``.  The output can be copy/pasted\n",
      "into a main LaTeX document or read from an external file\n",
      "with ``\\input{table.tex}``.\n",
      "\n",
      ".. versionchanged:: 1.0.0\n",
      "   Added caption and label arguments.\n",
      "\n",
      ".. versionchanged:: 1.2.0\n",
      "   Added position argument, changed meaning of caption argument.\n",
      "\n",
      "Parameters\n",
      "----------\n",
      "buf : str, Path or StringIO-like, optional, default None\n",
      "    Buffer to write to. If None, the output is returned as a string.\n",
      "columns : list of label, optional\n",
      "    The subset of columns to write. Writes all columns by default.\n",
      "col_space : int, optional\n",
      "    The minimum width of each column.\n",
      "header : bool or list of str, default True\n",
      "    Write out the column names. If a list of strings is given,\n",
      "    it is assumed to be aliases for the column names.\n",
      "index : bool, default True\n",
      "    Write row names (index).\n",
      "na_rep : str, default 'NaN'\n",
      "    Missing data representation.\n",
      "formatters : list of functions or dict of {str: function}, optional\n",
      "    Formatter functions to apply to columns' elements by position or\n",
      "    name. The result of each function must be a unicode string.\n",
      "    List must be of length equal to the number of columns.\n",
      "float_format : one-parameter function or str, optional, default None\n",
      "    Formatter for floating point numbers. For example\n",
      "    ``float_format=\"%.2f\"`` and ``float_format=\"{:0.2f}\".format`` will\n",
      "    both result in 0.1234 being formatted as 0.12.\n",
      "sparsify : bool, optional\n",
      "    Set to False for a DataFrame with a hierarchical index to print\n",
      "    every multiindex key at each row. By default, the value will be\n",
      "    read from the config module.\n",
      "index_names : bool, default True\n",
      "    Prints the names of the indexes.\n",
      "bold_rows : bool, default False\n",
      "    Make the row labels bold in the output.\n",
      "column_format : str, optional\n",
      "    The columns format as specified in `LaTeX table format\n",
      "    <https://en.wikibooks.org/wiki/LaTeX/Tables>`__ e.g. 'rcl' for 3\n",
      "    columns. By default, 'l' will be used for all columns except\n",
      "    columns of numbers, which default to 'r'.\n",
      "longtable : bool, optional\n",
      "    By default, the value will be read from the pandas config\n",
      "    module. Use a longtable environment instead of tabular. Requires\n",
      "    adding a \\usepackage{longtable} to your LaTeX preamble.\n",
      "escape : bool, optional\n",
      "    By default, the value will be read from the pandas config\n",
      "    module. When set to False prevents from escaping latex special\n",
      "    characters in column names.\n",
      "encoding : str, optional\n",
      "    A string representing the encoding to use in the output file,\n",
      "    defaults to 'utf-8'.\n",
      "decimal : str, default '.'\n",
      "    Character recognized as decimal separator, e.g. ',' in Europe.\n",
      "multicolumn : bool, default True\n",
      "    Use \\multicolumn to enhance MultiIndex columns.\n",
      "    The default will be read from the config module.\n",
      "multicolumn_format : str, default 'l'\n",
      "    The alignment for multicolumns, similar to `column_format`\n",
      "    The default will be read from the config module.\n",
      "multirow : bool, default False\n",
      "    Use \\multirow to enhance MultiIndex rows. Requires adding a\n",
      "    \\usepackage{multirow} to your LaTeX preamble. Will print\n",
      "    centered labels (instead of top-aligned) across the contained\n",
      "    rows, separating groups via clines. The default will be read\n",
      "    from the pandas config module.\n",
      "caption : str or tuple, optional\n",
      "    Tuple (full_caption, short_caption),\n",
      "    which results in ``\\caption[short_caption]{full_caption}``;\n",
      "    if a single string is passed, no short caption will be set.\n",
      "\n",
      "    .. versionadded:: 1.0.0\n",
      "\n",
      "    .. versionchanged:: 1.2.0\n",
      "       Optionally allow caption to be a tuple ``(full_caption, short_caption)``.\n",
      "\n",
      "label : str, optional\n",
      "    The LaTeX label to be placed inside ``\\label{}`` in the output.\n",
      "    This is used with ``\\ref{}`` in the main ``.tex`` file.\n",
      "\n",
      "    .. versionadded:: 1.0.0\n",
      "position : str, optional\n",
      "    The LaTeX positional argument for tables, to be placed after\n",
      "    ``\\begin{}`` in the output.\n",
      "\n",
      "    .. versionadded:: 1.2.0\n",
      "\n",
      "        Returns\n",
      "        -------\n",
      "        str or None\n",
      "            If buf is None, returns the result as a string. Otherwise returns\n",
      "            None.\n",
      "    \n",
      "See Also\n",
      "--------\n",
      "Styler.to_latex : Render a DataFrame to LaTeX with conditional formatting.\n",
      "DataFrame.to_string : Render a DataFrame to a console-friendly\n",
      "    tabular output.\n",
      "DataFrame.to_html : Render a DataFrame as an HTML table.\n",
      "\n",
      "Examples\n",
      "--------\n",
      ">>> df = pd.DataFrame(dict(name=['Raphael', 'Donatello'],\n",
      "...                   mask=['red', 'purple'],\n",
      "...                   weapon=['sai', 'bo staff']))\n",
      ">>> print(df.to_latex(index=False))  # doctest: +SKIP\n",
      "\\begin{tabular}{lll}\n",
      " \\toprule\n",
      "       name &    mask &    weapon \\\\\n",
      " \\midrule\n",
      "    Raphael &     red &       sai \\\\\n",
      "  Donatello &  purple &  bo staff \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\u001b[0;31mFile:\u001b[0m      ~/mambaforge/envs/bigbio/lib/python3.9/site-packages/pandas/core/generic.py\n",
      "\u001b[0;31mType:\u001b[0m      function\n"
     ]
    }
   ],
   "source": [
    "pd.DataFrame.to_latex?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Convert .csv into Markdown table about models\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare statistics of all datasets in BigBio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "ner_helpers = conhelps.filtered(\n",
    "    lambda x:\n",
    "        x.is_bigbio_schema\n",
    "        and x.is_local\n",
    "        and \"NAMED_ENTITY_DISAMBIGUATION\" in x.tasks\n",
    "        and not x.is_large\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('msh_wsd', 'msh_wsd_bigbio_kb', {'NAMED_ENTITY_DISAMBIGUATION'}),\n",
       " ('nlm_wsd',\n",
       "  'nlm_wsd_non_reviewed_bigbio_kb',\n",
       "  {'NAMED_ENTITY_DISAMBIGUATION'}),\n",
       " ('nlm_wsd', 'nlm_wsd_reviewed_bigbio_kb', {'NAMED_ENTITY_DISAMBIGUATION'})]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[(ch.dataset_name, ch.config.name, ch.tasks) for ch in ner_helpers]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.027201175689697266,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "Downloading builder script",
       "rate": null,
       "total": 9933,
       "unit": "B",
       "unit_divisor": 1000,
       "unit_scale": true
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "aa8b3b94b86944af9f674c653f4b9f0e",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading builder script:   0%|          | 0.00/9.93k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.02729511260986328,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "Downloading readme",
       "rate": null,
       "total": 1741,
       "unit": "B",
       "unit_divisor": 1000,
       "unit_scale": true
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "338539ce99ad4a34997905a2893fc3f5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading readme:   0%|          | 0.00/1.74k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.025624513626098633,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "Downloading extra modules",
       "rate": null,
       "total": 19286,
       "unit": "B",
       "unit_divisor": 1000,
       "unit_scale": true
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0bb4daba546f41f8b7471a869c2fd844",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading extra modules:   0%|          | 0.00/19.3k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading and preparing dataset msh_wsd/msh_wsd_bigbio_kb to /nethome/dkartchner3/.cache/huggingface/datasets/bigbio___msh_wsd/msh_wsd_bigbio_kb/1.0.0/21843cdf3630cd9ffb7dfd324f296850bddfdf1c9cb125746eba43ccecb97116...\n",
      "Exception!!!!\n",
      "This is a local dataset. Please pass the data_dir kwarg to load_dataset.\n"
     ]
    },
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.02548956871032715,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "Downloading builder script",
       "rate": null,
       "total": 14483,
       "unit": "B",
       "unit_divisor": 1000,
       "unit_scale": true
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "c826e4f406d24a019ba8dfddd7a820eb",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading builder script:   0%|          | 0.00/14.5k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.02670454978942871,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "Downloading readme",
       "rate": null,
       "total": 1768,
       "unit": "B",
       "unit_divisor": 1000,
       "unit_scale": true
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1fecafe604c44a2086cd4671056cc75d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading readme:   0%|          | 0.00/1.77k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.024033784866333008,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "Downloading extra modules",
       "rate": null,
       "total": 19286,
       "unit": "B",
       "unit_divisor": 1000,
       "unit_scale": true
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f04e71d164f94d889c77431ad69609bb",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading extra modules:   0%|          | 0.00/19.3k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading and preparing dataset nlm_wsd/nlm_wsd_non_reviewed_bigbio_kb to /nethome/dkartchner3/.cache/huggingface/datasets/bigbio___nlm_wsd/nlm_wsd_non_reviewed_bigbio_kb/1.0.0/d24f852bf92e5ec11cbf0973a22d3d58b88550a01b8e70295c169f324ae18d66...\n",
      "Exception!!!!\n",
      "This is a local dataset. Please pass the data_dir kwarg to load_dataset.\n",
      "Downloading and preparing dataset nlm_wsd/nlm_wsd_reviewed_bigbio_kb to /nethome/dkartchner3/.cache/huggingface/datasets/bigbio___nlm_wsd/nlm_wsd_reviewed_bigbio_kb/1.0.0/d24f852bf92e5ec11cbf0973a22d3d58b88550a01b8e70295c169f324ae18d66...\n",
      "Exception!!!!\n",
      "This is a local dataset. Please pass the data_dir kwarg to load_dataset.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_statistics = []\n",
    "for ch in ner_helpers:\n",
    "    try:\n",
    "        metadata = ner_helpers.for_config_name(ch.config.name).get_metadata()\n",
    "        for split in metadata.keys():\n",
    "            s = metadata[split].__dict__\n",
    "            s['split'] = split\n",
    "            s['config_name'] = ch.config.name\n",
    "            s['dataset_name'] = ch.dataset_name\n",
    "            s['display_name'] = ch.display_name\n",
    "            all_statistics.append(s)\n",
    "    except Exception as e:\n",
    "        print(\"Exception!!!!\")\n",
    "        print(e)\n",
    "\n",
    "df = pd.DataFrame(all_statistics)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>display_name</th>\n",
       "      <th>dataset_name</th>\n",
       "      <th>samples_count</th>\n",
       "      <th>passages_count</th>\n",
       "      <th>passages_char_count</th>\n",
       "      <th>passages_type_counter</th>\n",
       "      <th>entities_count</th>\n",
       "      <th>entities_normalized_count</th>\n",
       "      <th>entities_type_counter</th>\n",
       "      <th>entities_db_name_counter</th>\n",
       "      <th>entities_unique_db_ids_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>config_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ask_a_patient_bigbio_kb</th>\n",
       "      <td>AskAPatient</td>\n",
       "      <td>ask_a_patient</td>\n",
       "      <td>173240</td>\n",
       "      <td>173240</td>\n",
       "      <td>2238807</td>\n",
       "      <td>{'social_media_text': 173240}</td>\n",
       "      <td>173240</td>\n",
       "      <td>173240</td>\n",
       "      <td>{'social_media_text': 173240}</td>\n",
       "      <td>{'SNOMED-CT|AMT': 173240}</td>\n",
       "      <td>16548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bc5cdr_bigbio_kb</th>\n",
       "      <td>BC5CDR</td>\n",
       "      <td>bc5cdr</td>\n",
       "      <td>1500</td>\n",
       "      <td>3000</td>\n",
       "      <td>1975456</td>\n",
       "      <td>{'title': 1500, 'abstract': 1500}</td>\n",
       "      <td>29271</td>\n",
       "      <td>29335</td>\n",
       "      <td>{'Chemical': 15953, 'Disease': 13318}</td>\n",
       "      <td>{'MESH': 29335}</td>\n",
       "      <td>3907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bionlp_st_2019_bb_bigbio_kb</th>\n",
       "      <td>BioNLP 2019 BB</td>\n",
       "      <td>bionlp_st_2019_bb</td>\n",
       "      <td>295</td>\n",
       "      <td>295</td>\n",
       "      <td>305827</td>\n",
       "      <td>{'abstract': 295}</td>\n",
       "      <td>3456</td>\n",
       "      <td>3628</td>\n",
       "      <td>{'Habitat': 1752, 'Microorganism': 1174, 'Phen...</td>\n",
       "      <td>{'OntoBiotope': 2452, 'NCBI_Taxonomy': 1176}</td>\n",
       "      <td>834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>biorelex_bigbio_kb</th>\n",
       "      <td>BioRelEx</td>\n",
       "      <td>biorelex</td>\n",
       "      <td>1606</td>\n",
       "      <td>1606</td>\n",
       "      <td>270233</td>\n",
       "      <td>{'sentence': 1606}</td>\n",
       "      <td>7880</td>\n",
       "      <td>10593</td>\n",
       "      <td>{'protein': 3730, 'protein-family': 1125, 'che...</td>\n",
       "      <td>{'uniprot': 3190, 'hgnc': 2947, 'NCBI gene': 2...</td>\n",
       "      <td>3291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cadec_bigbio_kb</th>\n",
       "      <td>CADEC</td>\n",
       "      <td>cadec</td>\n",
       "      <td>1250</td>\n",
       "      <td>1250</td>\n",
       "      <td>575453</td>\n",
       "      <td>{'abstract': 1250}</td>\n",
       "      <td>9111</td>\n",
       "      <td>15932</td>\n",
       "      <td>{'ADR': 6318, 'Drug': 1800, 'Finding': 435, 'D...</td>\n",
       "      <td>{'Snomed CT': 8930, 'Meddra': 6222, '': 780}</td>\n",
       "      <td>1736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cantemist_bigbio_kb</th>\n",
       "      <td>CANTEMIST</td>\n",
       "      <td>cantemist</td>\n",
       "      <td>1301</td>\n",
       "      <td>1301</td>\n",
       "      <td>6344976</td>\n",
       "      <td>{'abstract': 1301}</td>\n",
       "      <td>16032</td>\n",
       "      <td>16032</td>\n",
       "      <td>{'MORFOLOGIA_NEOPLASIA': 16032}</td>\n",
       "      <td>{'eCIE-O-3.1': 16032}</td>\n",
       "      <td>1399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>citation_gia_test_collection_bigbio_kb</th>\n",
       "      <td>Citation GIA Test Collection</td>\n",
       "      <td>citation_gia_test_collection</td>\n",
       "      <td>151</td>\n",
       "      <td>302</td>\n",
       "      <td>229882</td>\n",
       "      <td>{'title': 151, 'abstract': 151}</td>\n",
       "      <td>1382</td>\n",
       "      <td>1382</td>\n",
       "      <td>{'Gene': 1205, 'FamilyName': 160, 'DomainMotif...</td>\n",
       "      <td>{'': 1382}</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>codiesp_X_bigbio_kb</th>\n",
       "      <td>CodiEsp</td>\n",
       "      <td>codiesp</td>\n",
       "      <td>1000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>{}</td>\n",
       "      <td>18435</td>\n",
       "      <td>18435</td>\n",
       "      <td>{'DIAGNOSTICO': 14305, 'PROCEDIMIENTO': 4130}</td>\n",
       "      <td>{'ICD10-CM': 14305, 'ICD10-PCS': 4130}</td>\n",
       "      <td>5377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distemist_entities_bigbio_kb</th>\n",
       "      <td>DisTEMIST</td>\n",
       "      <td>distemist</td>\n",
       "      <td>750</td>\n",
       "      <td>750</td>\n",
       "      <td>1758414</td>\n",
       "      <td>{'clinical_case': 750}</td>\n",
       "      <td>8065</td>\n",
       "      <td>0</td>\n",
       "      <td>{'ENFERMEDAD': 8065}</td>\n",
       "      <td>{}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distemist_linking_bigbio_kb</th>\n",
       "      <td>DisTEMIST</td>\n",
       "      <td>distemist</td>\n",
       "      <td>583</td>\n",
       "      <td>583</td>\n",
       "      <td>1246383</td>\n",
       "      <td>{'clinical_case': 583}</td>\n",
       "      <td>5136</td>\n",
       "      <td>5374</td>\n",
       "      <td>{'ENFERMEDAD': 5136}</td>\n",
       "      <td>{'SNOMED_CT': 5374}</td>\n",
       "      <td>2430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gnormplus_bigbio_kb</th>\n",
       "      <td>GNormPlus</td>\n",
       "      <td>gnormplus</td>\n",
       "      <td>543</td>\n",
       "      <td>1086</td>\n",
       "      <td>737897</td>\n",
       "      <td>{'title': 543, 'abstract': 543}</td>\n",
       "      <td>9257</td>\n",
       "      <td>6430</td>\n",
       "      <td>{'Gene': 6251, 'FamilyName': 2367, 'DomainMoti...</td>\n",
       "      <td>{'NCBIGene': 6430}</td>\n",
       "      <td>1465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>linnaeus_bigbio_kb</th>\n",
       "      <td>LINNAEUS</td>\n",
       "      <td>linnaeus</td>\n",
       "      <td>95</td>\n",
       "      <td>95</td>\n",
       "      <td>2612606</td>\n",
       "      <td>{'Article': 95}</td>\n",
       "      <td>4259</td>\n",
       "      <td>4259</td>\n",
       "      <td>{'species': 4259}</td>\n",
       "      <td>{'ncbi': 4259}</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>linnaeus_filtered_bigbio_kb</th>\n",
       "      <td>LINNAEUS</td>\n",
       "      <td>linnaeus</td>\n",
       "      <td>84</td>\n",
       "      <td>84</td>\n",
       "      <td>2464729</td>\n",
       "      <td>{'Article': 84}</td>\n",
       "      <td>2851</td>\n",
       "      <td>2851</td>\n",
       "      <td>{'species': 2851}</td>\n",
       "      <td>{'ncbi': 2851}</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_de_emea_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>12961</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>425</td>\n",
       "      <td>425</td>\n",
       "      <td>{'DISO': 151, 'CHEM': 99, 'PROC': 57, 'LIVB': ...</td>\n",
       "      <td>{'UMLS': 425}</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_de_medline_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>7606</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>309</td>\n",
       "      <td>309</td>\n",
       "      <td>{'DISO': 77, 'PROC': 71, 'ANAT': 56, 'CHEM': 5...</td>\n",
       "      <td>{'UMLS': 309}</td>\n",
       "      <td>259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_de_patents_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>17419</td>\n",
       "      <td>{'abstract': 50}</td>\n",
       "      <td>348</td>\n",
       "      <td>348</td>\n",
       "      <td>{'CHEM': 149, 'DISO': 101, 'ANAT': 31, 'PROC':...</td>\n",
       "      <td>{'UMLS': 348}</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_en_emea_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>10958</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>433</td>\n",
       "      <td>433</td>\n",
       "      <td>{'DISO': 150, 'CHEM': 100, 'PROC': 62, 'LIVB':...</td>\n",
       "      <td>{'UMLS': 433}</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_en_medline_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>400</td>\n",
       "      <td>400</td>\n",
       "      <td>27821</td>\n",
       "      <td>{'abstract': 400}</td>\n",
       "      <td>1177</td>\n",
       "      <td>1177</td>\n",
       "      <td>{'DISO': 359, 'PROC': 250, 'LIVB': 157, 'ANAT'...</td>\n",
       "      <td>{'UMLS': 1177}</td>\n",
       "      <td>884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_en_patents_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>16021</td>\n",
       "      <td>{'abstract': 50}</td>\n",
       "      <td>353</td>\n",
       "      <td>353</td>\n",
       "      <td>{'CHEM': 150, 'DISO': 104, 'ANAT': 31, 'PROC':...</td>\n",
       "      <td>{'UMLS': 353}</td>\n",
       "      <td>227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_es_emea_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>12490</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>432</td>\n",
       "      <td>432</td>\n",
       "      <td>{'DISO': 151, 'CHEM': 98, 'PROC': 66, 'LIVB': ...</td>\n",
       "      <td>{'UMLS': 432}</td>\n",
       "      <td>295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_es_medline_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>7953</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>324</td>\n",
       "      <td>324</td>\n",
       "      <td>{'DISO': 115, 'PROC': 63, 'LIVB': 42, 'ANAT': ...</td>\n",
       "      <td>{'UMLS': 324}</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_fr_emea_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>13331</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>431</td>\n",
       "      <td>431</td>\n",
       "      <td>{'DISO': 151, 'CHEM': 101, 'PROC': 59, 'LIVB':...</td>\n",
       "      <td>{'UMLS': 431}</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_fr_medline_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>7642</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>279</td>\n",
       "      <td>279</td>\n",
       "      <td>{'DISO': 86, 'PROC': 64, 'LIVB': 45, 'ANAT': 3...</td>\n",
       "      <td>{'UMLS': 279}</td>\n",
       "      <td>242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_fr_patents_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>18123</td>\n",
       "      <td>{'abstract': 50}</td>\n",
       "      <td>342</td>\n",
       "      <td>342</td>\n",
       "      <td>{'CHEM': 146, 'DISO': 103, 'ANAT': 31, 'PROC':...</td>\n",
       "      <td>{'UMLS': 342}</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_nl_emea_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>12370</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>434</td>\n",
       "      <td>434</td>\n",
       "      <td>{'DISO': 159, 'CHEM': 101, 'PROC': 57, 'LIVB':...</td>\n",
       "      <td>{'UMLS': 434}</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mantra_gsc_nl_medline_bigbio_kb</th>\n",
       "      <td>Mantra GSC</td>\n",
       "      <td>mantra_gsc</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>6332</td>\n",
       "      <td>{'abstract': 100}</td>\n",
       "      <td>243</td>\n",
       "      <td>243</td>\n",
       "      <td>{'DISO': 74, 'PROC': 49, 'LIVB': 49, 'ANAT': 2...</td>\n",
       "      <td>{'UMLS': 243}</td>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>medmentions_full_bigbio_kb</th>\n",
       "      <td>MedMentions</td>\n",
       "      <td>medmentions</td>\n",
       "      <td>4392</td>\n",
       "      <td>8784</td>\n",
       "      <td>6945085</td>\n",
       "      <td>{'title': 4392, 'abstract': 4392}</td>\n",
       "      <td>385098</td>\n",
       "      <td>385098</td>\n",
       "      <td>{'T080': 31485, 'T169': 23661, 'T081': 19995, ...</td>\n",
       "      <td>{'UMLS': 385098}</td>\n",
       "      <td>50720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>medmentions_st21pv_bigbio_kb</th>\n",
       "      <td>MedMentions</td>\n",
       "      <td>medmentions</td>\n",
       "      <td>4392</td>\n",
       "      <td>8784</td>\n",
       "      <td>6945085</td>\n",
       "      <td>{'title': 4392, 'abstract': 4392}</td>\n",
       "      <td>203282</td>\n",
       "      <td>203282</td>\n",
       "      <td>{'T038': 41422, 'T103': 37401, 'T058': 24306, ...</td>\n",
       "      <td>{'UMLS': 203282}</td>\n",
       "      <td>35620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mirna_bigbio_kb</th>\n",
       "      <td>miRNA</td>\n",
       "      <td>mirna</td>\n",
       "      <td>301</td>\n",
       "      <td>2411</td>\n",
       "      <td>385368</td>\n",
       "      <td>{'abstract': 2133, 'title': 278}</td>\n",
       "      <td>8318</td>\n",
       "      <td>8318</td>\n",
       "      <td>{'Diseases': 2161, 'Relation_Trigger': 1960, '...</td>\n",
       "      <td>{'miRNA-corpus': 8318}</td>\n",
       "      <td>8315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>muchmore_bigbio_kb</th>\n",
       "      <td>MuchMore</td>\n",
       "      <td>muchmore</td>\n",
       "      <td>15630</td>\n",
       "      <td>15630</td>\n",
       "      <td>16009519</td>\n",
       "      <td>{'abstract': 15630}</td>\n",
       "      <td>345838</td>\n",
       "      <td>399189</td>\n",
       "      <td>{'umlsterm': 345838}</td>\n",
       "      <td>{'UMLS': 399189}</td>\n",
       "      <td>10004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>muchmore_de_bigbio_kb</th>\n",
       "      <td>MuchMore</td>\n",
       "      <td>muchmore</td>\n",
       "      <td>7808</td>\n",
       "      <td>7808</td>\n",
       "      <td>8425551</td>\n",
       "      <td>{'abstract': 7808}</td>\n",
       "      <td>112596</td>\n",
       "      <td>121306</td>\n",
       "      <td>{'umlsterm': 112596}</td>\n",
       "      <td>{'UMLS': 121306}</td>\n",
       "      <td>5573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>muchmore_en_bigbio_kb</th>\n",
       "      <td>MuchMore</td>\n",
       "      <td>muchmore</td>\n",
       "      <td>7822</td>\n",
       "      <td>7822</td>\n",
       "      <td>7583968</td>\n",
       "      <td>{'abstract': 7822}</td>\n",
       "      <td>233242</td>\n",
       "      <td>277883</td>\n",
       "      <td>{'umlsterm': 233242}</td>\n",
       "      <td>{'UMLS': 277883}</td>\n",
       "      <td>9176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncbi_disease_bigbio_kb</th>\n",
       "      <td>NCBI Disease</td>\n",
       "      <td>ncbi_disease</td>\n",
       "      <td>792</td>\n",
       "      <td>1584</td>\n",
       "      <td>1014349</td>\n",
       "      <td>{'title': 792, 'abstract': 792}</td>\n",
       "      <td>6881</td>\n",
       "      <td>7054</td>\n",
       "      <td>{'SpecificDisease': 3929, 'Modifier': 1766, 'D...</td>\n",
       "      <td>{'MESH': 6487, 'OMIM': 567}</td>\n",
       "      <td>1015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nlm_gene_bigbio_kb</th>\n",
       "      <td>NLM-Gene</td>\n",
       "      <td>nlm_gene</td>\n",
       "      <td>550</td>\n",
       "      <td>1100</td>\n",
       "      <td>991536</td>\n",
       "      <td>{'title': 550, 'abstract': 550}</td>\n",
       "      <td>15553</td>\n",
       "      <td>18056</td>\n",
       "      <td>{'Gene': 9562, 'GENERIF': 3887, 'STARGENE': 19...</td>\n",
       "      <td>{'NCBIGene': 18056}</td>\n",
       "      <td>3940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nlmchem_bigbio_kb</th>\n",
       "      <td>NLM-Chem</td>\n",
       "      <td>nlmchem</td>\n",
       "      <td>150</td>\n",
       "      <td>10310</td>\n",
       "      <td>4861074</td>\n",
       "      <td>{'paragraph': 5136, 'title_2': 1549, 'fig_capt...</td>\n",
       "      <td>38339</td>\n",
       "      <td>41300</td>\n",
       "      <td>{'Chemical': 38339}</td>\n",
       "      <td>{'MESH': 41300}</td>\n",
       "      <td>2486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>osiris_bigbio_kb</th>\n",
       "      <td>OSIRIS</td>\n",
       "      <td>osiris</td>\n",
       "      <td>105</td>\n",
       "      <td>210</td>\n",
       "      <td>172056</td>\n",
       "      <td>{'title': 105, 'abstract': 105}</td>\n",
       "      <td>1230</td>\n",
       "      <td>1230</td>\n",
       "      <td>{'gene': 689, 'variant': 541}</td>\n",
       "      <td>{'NCBI Gene': 689, 'HGVS-like': 291, 'dbSNP': ...</td>\n",
       "      <td>423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pubtator_central_sample_bigbio_kb</th>\n",
       "      <td>PubTator Central</td>\n",
       "      <td>pubtator_central</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>19454</td>\n",
       "      <td>{'title': 4, 'abstract': 4}</td>\n",
       "      <td>208</td>\n",
       "      <td>208</td>\n",
       "      <td>{'Disease': 77, 'Chemical': 62, 'Species': 49,...</td>\n",
       "      <td>{'mesh': 139, 'ncbi_taxon': 49, 'ncbi_gene': 1...</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>quaero_emea_bigbio_kb</th>\n",
       "      <td>QUAERO</td>\n",
       "      <td>quaero</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>227811</td>\n",
       "      <td>{'abstract': 38}</td>\n",
       "      <td>7159</td>\n",
       "      <td>7202</td>\n",
       "      <td>{'DISO': 1510, 'CHEM': 2482, 'PROC': 952, 'LIV...</td>\n",
       "      <td>{'UMLS': 7202}</td>\n",
       "      <td>1643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>quaero_medline_bigbio_kb</th>\n",
       "      <td>QUAERO</td>\n",
       "      <td>quaero</td>\n",
       "      <td>2498</td>\n",
       "      <td>2498</td>\n",
       "      <td>205903</td>\n",
       "      <td>{'abstract': 2498}</td>\n",
       "      <td>9074</td>\n",
       "      <td>9079</td>\n",
       "      <td>{'DISO': 2843, 'PROC': 1750, 'ANAT': 1499, 'CH...</td>\n",
       "      <td>{'UMLS': 9079}</td>\n",
       "      <td>5616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spl_adr_200db_train_bigbio_kb</th>\n",
       "      <td>SPL ADR</td>\n",
       "      <td>spl_adr_200db</td>\n",
       "      <td>101</td>\n",
       "      <td>239</td>\n",
       "      <td>1769683</td>\n",
       "      <td>{'adverse reactions': 101, 'warnings and preca...</td>\n",
       "      <td>15722</td>\n",
       "      <td>2417</td>\n",
       "      <td>{'AdverseReaction': 13795, 'Severity': 934, 'F...</td>\n",
       "      <td>{'MedDRA v18.1': 2417}</td>\n",
       "      <td>637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spl_adr_200db_unannotated_bigbio_kb</th>\n",
       "      <td>SPL ADR</td>\n",
       "      <td>spl_adr_200db</td>\n",
       "      <td>2208</td>\n",
       "      <td>4311</td>\n",
       "      <td>28952558</td>\n",
       "      <td>{'adverse reactions': 2267, 'warnings and prec...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>{}</td>\n",
       "      <td>{}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>thomas2011_bigbio_kb</th>\n",
       "      <td>SNP Corpus</td>\n",
       "      <td>thomas2011</td>\n",
       "      <td>296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>{}</td>\n",
       "      <td>527</td>\n",
       "      <td>527</td>\n",
       "      <td>{'Protein Sequence Mutation': 283, 'Nucleotide...</td>\n",
       "      <td>{'dbSNP': 527}</td>\n",
       "      <td>385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tmvar_v2_bigbio_kb</th>\n",
       "      <td>tmVar v2</td>\n",
       "      <td>tmvar_v2</td>\n",
       "      <td>158</td>\n",
       "      <td>316</td>\n",
       "      <td>258660</td>\n",
       "      <td>{'title': 158, 'abstract': 158}</td>\n",
       "      <td>674</td>\n",
       "      <td>625</td>\n",
       "      <td>{'ProteinMutation': 436, 'DNAMutation': 190, '...</td>\n",
       "      <td>{'dbsnp': 625}</td>\n",
       "      <td>275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tmvar_v3_bigbio_kb</th>\n",
       "      <td>tmVar v3</td>\n",
       "      <td>tmvar_v3</td>\n",
       "      <td>500</td>\n",
       "      <td>1000</td>\n",
       "      <td>811380</td>\n",
       "      <td>{'title': 500, 'abstract': 500}</td>\n",
       "      <td>7837</td>\n",
       "      <td>8884</td>\n",
       "      <td>{'['Gene']': 4059, '['Species']': 1756, '['Pro...</td>\n",
       "      <td>{'NCBI Gene': 6023, 'dbSNP': 2861}</td>\n",
       "      <td>2161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>twadrl_bigbio_kb</th>\n",
       "      <td>TwADR-L</td>\n",
       "      <td>twadrl</td>\n",
       "      <td>50740</td>\n",
       "      <td>50740</td>\n",
       "      <td>799570</td>\n",
       "      <td>{'social_media_text': 50740}</td>\n",
       "      <td>50740</td>\n",
       "      <td>50740</td>\n",
       "      <td>{'social_media_text': 50740}</td>\n",
       "      <td>{'SIDER 4': 50740}</td>\n",
       "      <td>23819</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        display_name  \\\n",
       "config_name                                                            \n",
       "ask_a_patient_bigbio_kb                                  AskAPatient   \n",
       "bc5cdr_bigbio_kb                                              BC5CDR   \n",
       "bionlp_st_2019_bb_bigbio_kb                           BioNLP 2019 BB   \n",
       "biorelex_bigbio_kb                                          BioRelEx   \n",
       "cadec_bigbio_kb                                                CADEC   \n",
       "cantemist_bigbio_kb                                        CANTEMIST   \n",
       "citation_gia_test_collection_bigbio_kb  Citation GIA Test Collection   \n",
       "codiesp_X_bigbio_kb                                          CodiEsp   \n",
       "distemist_entities_bigbio_kb                               DisTEMIST   \n",
       "distemist_linking_bigbio_kb                                DisTEMIST   \n",
       "gnormplus_bigbio_kb                                        GNormPlus   \n",
       "linnaeus_bigbio_kb                                          LINNAEUS   \n",
       "linnaeus_filtered_bigbio_kb                                 LINNAEUS   \n",
       "mantra_gsc_de_emea_bigbio_kb                              Mantra GSC   \n",
       "mantra_gsc_de_medline_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_de_patents_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_en_emea_bigbio_kb                              Mantra GSC   \n",
       "mantra_gsc_en_medline_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_en_patents_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_es_emea_bigbio_kb                              Mantra GSC   \n",
       "mantra_gsc_es_medline_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_fr_emea_bigbio_kb                              Mantra GSC   \n",
       "mantra_gsc_fr_medline_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_fr_patents_bigbio_kb                           Mantra GSC   \n",
       "mantra_gsc_nl_emea_bigbio_kb                              Mantra GSC   \n",
       "mantra_gsc_nl_medline_bigbio_kb                           Mantra GSC   \n",
       "medmentions_full_bigbio_kb                               MedMentions   \n",
       "medmentions_st21pv_bigbio_kb                             MedMentions   \n",
       "mirna_bigbio_kb                                                miRNA   \n",
       "muchmore_bigbio_kb                                          MuchMore   \n",
       "muchmore_de_bigbio_kb                                       MuchMore   \n",
       "muchmore_en_bigbio_kb                                       MuchMore   \n",
       "ncbi_disease_bigbio_kb                                  NCBI Disease   \n",
       "nlm_gene_bigbio_kb                                          NLM-Gene   \n",
       "nlmchem_bigbio_kb                                           NLM-Chem   \n",
       "osiris_bigbio_kb                                              OSIRIS   \n",
       "pubtator_central_sample_bigbio_kb                   PubTator Central   \n",
       "quaero_emea_bigbio_kb                                         QUAERO   \n",
       "quaero_medline_bigbio_kb                                      QUAERO   \n",
       "spl_adr_200db_train_bigbio_kb                                SPL ADR   \n",
       "spl_adr_200db_unannotated_bigbio_kb                          SPL ADR   \n",
       "thomas2011_bigbio_kb                                      SNP Corpus   \n",
       "tmvar_v2_bigbio_kb                                          tmVar v2   \n",
       "tmvar_v3_bigbio_kb                                          tmVar v3   \n",
       "twadrl_bigbio_kb                                             TwADR-L   \n",
       "\n",
       "                                                        dataset_name  \\\n",
       "config_name                                                            \n",
       "ask_a_patient_bigbio_kb                                ask_a_patient   \n",
       "bc5cdr_bigbio_kb                                              bc5cdr   \n",
       "bionlp_st_2019_bb_bigbio_kb                        bionlp_st_2019_bb   \n",
       "biorelex_bigbio_kb                                          biorelex   \n",
       "cadec_bigbio_kb                                                cadec   \n",
       "cantemist_bigbio_kb                                        cantemist   \n",
       "citation_gia_test_collection_bigbio_kb  citation_gia_test_collection   \n",
       "codiesp_X_bigbio_kb                                          codiesp   \n",
       "distemist_entities_bigbio_kb                               distemist   \n",
       "distemist_linking_bigbio_kb                                distemist   \n",
       "gnormplus_bigbio_kb                                        gnormplus   \n",
       "linnaeus_bigbio_kb                                          linnaeus   \n",
       "linnaeus_filtered_bigbio_kb                                 linnaeus   \n",
       "mantra_gsc_de_emea_bigbio_kb                              mantra_gsc   \n",
       "mantra_gsc_de_medline_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_de_patents_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_en_emea_bigbio_kb                              mantra_gsc   \n",
       "mantra_gsc_en_medline_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_en_patents_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_es_emea_bigbio_kb                              mantra_gsc   \n",
       "mantra_gsc_es_medline_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_fr_emea_bigbio_kb                              mantra_gsc   \n",
       "mantra_gsc_fr_medline_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_fr_patents_bigbio_kb                           mantra_gsc   \n",
       "mantra_gsc_nl_emea_bigbio_kb                              mantra_gsc   \n",
       "mantra_gsc_nl_medline_bigbio_kb                           mantra_gsc   \n",
       "medmentions_full_bigbio_kb                               medmentions   \n",
       "medmentions_st21pv_bigbio_kb                             medmentions   \n",
       "mirna_bigbio_kb                                                mirna   \n",
       "muchmore_bigbio_kb                                          muchmore   \n",
       "muchmore_de_bigbio_kb                                       muchmore   \n",
       "muchmore_en_bigbio_kb                                       muchmore   \n",
       "ncbi_disease_bigbio_kb                                  ncbi_disease   \n",
       "nlm_gene_bigbio_kb                                          nlm_gene   \n",
       "nlmchem_bigbio_kb                                            nlmchem   \n",
       "osiris_bigbio_kb                                              osiris   \n",
       "pubtator_central_sample_bigbio_kb                   pubtator_central   \n",
       "quaero_emea_bigbio_kb                                         quaero   \n",
       "quaero_medline_bigbio_kb                                      quaero   \n",
       "spl_adr_200db_train_bigbio_kb                          spl_adr_200db   \n",
       "spl_adr_200db_unannotated_bigbio_kb                    spl_adr_200db   \n",
       "thomas2011_bigbio_kb                                      thomas2011   \n",
       "tmvar_v2_bigbio_kb                                          tmvar_v2   \n",
       "tmvar_v3_bigbio_kb                                          tmvar_v3   \n",
       "twadrl_bigbio_kb                                              twadrl   \n",
       "\n",
       "                                        samples_count  passages_count  \\\n",
       "config_name                                                             \n",
       "ask_a_patient_bigbio_kb                        173240          173240   \n",
       "bc5cdr_bigbio_kb                                 1500            3000   \n",
       "bionlp_st_2019_bb_bigbio_kb                       295             295   \n",
       "biorelex_bigbio_kb                               1606            1606   \n",
       "cadec_bigbio_kb                                  1250            1250   \n",
       "cantemist_bigbio_kb                              1301            1301   \n",
       "citation_gia_test_collection_bigbio_kb            151             302   \n",
       "codiesp_X_bigbio_kb                              1000               0   \n",
       "distemist_entities_bigbio_kb                      750             750   \n",
       "distemist_linking_bigbio_kb                       583             583   \n",
       "gnormplus_bigbio_kb                               543            1086   \n",
       "linnaeus_bigbio_kb                                 95              95   \n",
       "linnaeus_filtered_bigbio_kb                        84              84   \n",
       "mantra_gsc_de_emea_bigbio_kb                      100             100   \n",
       "mantra_gsc_de_medline_bigbio_kb                   100             100   \n",
       "mantra_gsc_de_patents_bigbio_kb                    50              50   \n",
       "mantra_gsc_en_emea_bigbio_kb                      100             100   \n",
       "mantra_gsc_en_medline_bigbio_kb                   400             400   \n",
       "mantra_gsc_en_patents_bigbio_kb                    50              50   \n",
       "mantra_gsc_es_emea_bigbio_kb                      100             100   \n",
       "mantra_gsc_es_medline_bigbio_kb                   100             100   \n",
       "mantra_gsc_fr_emea_bigbio_kb                      100             100   \n",
       "mantra_gsc_fr_medline_bigbio_kb                   100             100   \n",
       "mantra_gsc_fr_patents_bigbio_kb                    50              50   \n",
       "mantra_gsc_nl_emea_bigbio_kb                      100             100   \n",
       "mantra_gsc_nl_medline_bigbio_kb                   100             100   \n",
       "medmentions_full_bigbio_kb                       4392            8784   \n",
       "medmentions_st21pv_bigbio_kb                     4392            8784   \n",
       "mirna_bigbio_kb                                   301            2411   \n",
       "muchmore_bigbio_kb                              15630           15630   \n",
       "muchmore_de_bigbio_kb                            7808            7808   \n",
       "muchmore_en_bigbio_kb                            7822            7822   \n",
       "ncbi_disease_bigbio_kb                            792            1584   \n",
       "nlm_gene_bigbio_kb                                550            1100   \n",
       "nlmchem_bigbio_kb                                 150           10310   \n",
       "osiris_bigbio_kb                                  105             210   \n",
       "pubtator_central_sample_bigbio_kb                   4               8   \n",
       "quaero_emea_bigbio_kb                              38              38   \n",
       "quaero_medline_bigbio_kb                         2498            2498   \n",
       "spl_adr_200db_train_bigbio_kb                     101             239   \n",
       "spl_adr_200db_unannotated_bigbio_kb              2208            4311   \n",
       "thomas2011_bigbio_kb                              296               0   \n",
       "tmvar_v2_bigbio_kb                                158             316   \n",
       "tmvar_v3_bigbio_kb                                500            1000   \n",
       "twadrl_bigbio_kb                                50740           50740   \n",
       "\n",
       "                                        passages_char_count  \\\n",
       "config_name                                                   \n",
       "ask_a_patient_bigbio_kb                             2238807   \n",
       "bc5cdr_bigbio_kb                                    1975456   \n",
       "bionlp_st_2019_bb_bigbio_kb                          305827   \n",
       "biorelex_bigbio_kb                                   270233   \n",
       "cadec_bigbio_kb                                      575453   \n",
       "cantemist_bigbio_kb                                 6344976   \n",
       "citation_gia_test_collection_bigbio_kb               229882   \n",
       "codiesp_X_bigbio_kb                                       0   \n",
       "distemist_entities_bigbio_kb                        1758414   \n",
       "distemist_linking_bigbio_kb                         1246383   \n",
       "gnormplus_bigbio_kb                                  737897   \n",
       "linnaeus_bigbio_kb                                  2612606   \n",
       "linnaeus_filtered_bigbio_kb                         2464729   \n",
       "mantra_gsc_de_emea_bigbio_kb                          12961   \n",
       "mantra_gsc_de_medline_bigbio_kb                        7606   \n",
       "mantra_gsc_de_patents_bigbio_kb                       17419   \n",
       "mantra_gsc_en_emea_bigbio_kb                          10958   \n",
       "mantra_gsc_en_medline_bigbio_kb                       27821   \n",
       "mantra_gsc_en_patents_bigbio_kb                       16021   \n",
       "mantra_gsc_es_emea_bigbio_kb                          12490   \n",
       "mantra_gsc_es_medline_bigbio_kb                        7953   \n",
       "mantra_gsc_fr_emea_bigbio_kb                          13331   \n",
       "mantra_gsc_fr_medline_bigbio_kb                        7642   \n",
       "mantra_gsc_fr_patents_bigbio_kb                       18123   \n",
       "mantra_gsc_nl_emea_bigbio_kb                          12370   \n",
       "mantra_gsc_nl_medline_bigbio_kb                        6332   \n",
       "medmentions_full_bigbio_kb                          6945085   \n",
       "medmentions_st21pv_bigbio_kb                        6945085   \n",
       "mirna_bigbio_kb                                      385368   \n",
       "muchmore_bigbio_kb                                 16009519   \n",
       "muchmore_de_bigbio_kb                               8425551   \n",
       "muchmore_en_bigbio_kb                               7583968   \n",
       "ncbi_disease_bigbio_kb                              1014349   \n",
       "nlm_gene_bigbio_kb                                   991536   \n",
       "nlmchem_bigbio_kb                                   4861074   \n",
       "osiris_bigbio_kb                                     172056   \n",
       "pubtator_central_sample_bigbio_kb                     19454   \n",
       "quaero_emea_bigbio_kb                                227811   \n",
       "quaero_medline_bigbio_kb                             205903   \n",
       "spl_adr_200db_train_bigbio_kb                       1769683   \n",
       "spl_adr_200db_unannotated_bigbio_kb                28952558   \n",
       "thomas2011_bigbio_kb                                      0   \n",
       "tmvar_v2_bigbio_kb                                   258660   \n",
       "tmvar_v3_bigbio_kb                                   811380   \n",
       "twadrl_bigbio_kb                                     799570   \n",
       "\n",
       "                                                                    passages_type_counter  \\\n",
       "config_name                                                                                 \n",
       "ask_a_patient_bigbio_kb                                     {'social_media_text': 173240}   \n",
       "bc5cdr_bigbio_kb                                        {'title': 1500, 'abstract': 1500}   \n",
       "bionlp_st_2019_bb_bigbio_kb                                             {'abstract': 295}   \n",
       "biorelex_bigbio_kb                                                     {'sentence': 1606}   \n",
       "cadec_bigbio_kb                                                        {'abstract': 1250}   \n",
       "cantemist_bigbio_kb                                                    {'abstract': 1301}   \n",
       "citation_gia_test_collection_bigbio_kb                    {'title': 151, 'abstract': 151}   \n",
       "codiesp_X_bigbio_kb                                                                    {}   \n",
       "distemist_entities_bigbio_kb                                       {'clinical_case': 750}   \n",
       "distemist_linking_bigbio_kb                                        {'clinical_case': 583}   \n",
       "gnormplus_bigbio_kb                                       {'title': 543, 'abstract': 543}   \n",
       "linnaeus_bigbio_kb                                                        {'Article': 95}   \n",
       "linnaeus_filtered_bigbio_kb                                               {'Article': 84}   \n",
       "mantra_gsc_de_emea_bigbio_kb                                            {'abstract': 100}   \n",
       "mantra_gsc_de_medline_bigbio_kb                                         {'abstract': 100}   \n",
       "mantra_gsc_de_patents_bigbio_kb                                          {'abstract': 50}   \n",
       "mantra_gsc_en_emea_bigbio_kb                                            {'abstract': 100}   \n",
       "mantra_gsc_en_medline_bigbio_kb                                         {'abstract': 400}   \n",
       "mantra_gsc_en_patents_bigbio_kb                                          {'abstract': 50}   \n",
       "mantra_gsc_es_emea_bigbio_kb                                            {'abstract': 100}   \n",
       "mantra_gsc_es_medline_bigbio_kb                                         {'abstract': 100}   \n",
       "mantra_gsc_fr_emea_bigbio_kb                                            {'abstract': 100}   \n",
       "mantra_gsc_fr_medline_bigbio_kb                                         {'abstract': 100}   \n",
       "mantra_gsc_fr_patents_bigbio_kb                                          {'abstract': 50}   \n",
       "mantra_gsc_nl_emea_bigbio_kb                                            {'abstract': 100}   \n",
       "mantra_gsc_nl_medline_bigbio_kb                                         {'abstract': 100}   \n",
       "medmentions_full_bigbio_kb                              {'title': 4392, 'abstract': 4392}   \n",
       "medmentions_st21pv_bigbio_kb                            {'title': 4392, 'abstract': 4392}   \n",
       "mirna_bigbio_kb                                          {'abstract': 2133, 'title': 278}   \n",
       "muchmore_bigbio_kb                                                    {'abstract': 15630}   \n",
       "muchmore_de_bigbio_kb                                                  {'abstract': 7808}   \n",
       "muchmore_en_bigbio_kb                                                  {'abstract': 7822}   \n",
       "ncbi_disease_bigbio_kb                                    {'title': 792, 'abstract': 792}   \n",
       "nlm_gene_bigbio_kb                                        {'title': 550, 'abstract': 550}   \n",
       "nlmchem_bigbio_kb                       {'paragraph': 5136, 'title_2': 1549, 'fig_capt...   \n",
       "osiris_bigbio_kb                                          {'title': 105, 'abstract': 105}   \n",
       "pubtator_central_sample_bigbio_kb                             {'title': 4, 'abstract': 4}   \n",
       "quaero_emea_bigbio_kb                                                    {'abstract': 38}   \n",
       "quaero_medline_bigbio_kb                                               {'abstract': 2498}   \n",
       "spl_adr_200db_train_bigbio_kb           {'adverse reactions': 101, 'warnings and preca...   \n",
       "spl_adr_200db_unannotated_bigbio_kb     {'adverse reactions': 2267, 'warnings and prec...   \n",
       "thomas2011_bigbio_kb                                                                   {}   \n",
       "tmvar_v2_bigbio_kb                                        {'title': 158, 'abstract': 158}   \n",
       "tmvar_v3_bigbio_kb                                        {'title': 500, 'abstract': 500}   \n",
       "twadrl_bigbio_kb                                             {'social_media_text': 50740}   \n",
       "\n",
       "                                        entities_count  \\\n",
       "config_name                                              \n",
       "ask_a_patient_bigbio_kb                         173240   \n",
       "bc5cdr_bigbio_kb                                 29271   \n",
       "bionlp_st_2019_bb_bigbio_kb                       3456   \n",
       "biorelex_bigbio_kb                                7880   \n",
       "cadec_bigbio_kb                                   9111   \n",
       "cantemist_bigbio_kb                              16032   \n",
       "citation_gia_test_collection_bigbio_kb            1382   \n",
       "codiesp_X_bigbio_kb                              18435   \n",
       "distemist_entities_bigbio_kb                      8065   \n",
       "distemist_linking_bigbio_kb                       5136   \n",
       "gnormplus_bigbio_kb                               9257   \n",
       "linnaeus_bigbio_kb                                4259   \n",
       "linnaeus_filtered_bigbio_kb                       2851   \n",
       "mantra_gsc_de_emea_bigbio_kb                       425   \n",
       "mantra_gsc_de_medline_bigbio_kb                    309   \n",
       "mantra_gsc_de_patents_bigbio_kb                    348   \n",
       "mantra_gsc_en_emea_bigbio_kb                       433   \n",
       "mantra_gsc_en_medline_bigbio_kb                   1177   \n",
       "mantra_gsc_en_patents_bigbio_kb                    353   \n",
       "mantra_gsc_es_emea_bigbio_kb                       432   \n",
       "mantra_gsc_es_medline_bigbio_kb                    324   \n",
       "mantra_gsc_fr_emea_bigbio_kb                       431   \n",
       "mantra_gsc_fr_medline_bigbio_kb                    279   \n",
       "mantra_gsc_fr_patents_bigbio_kb                    342   \n",
       "mantra_gsc_nl_emea_bigbio_kb                       434   \n",
       "mantra_gsc_nl_medline_bigbio_kb                    243   \n",
       "medmentions_full_bigbio_kb                      385098   \n",
       "medmentions_st21pv_bigbio_kb                    203282   \n",
       "mirna_bigbio_kb                                   8318   \n",
       "muchmore_bigbio_kb                              345838   \n",
       "muchmore_de_bigbio_kb                           112596   \n",
       "muchmore_en_bigbio_kb                           233242   \n",
       "ncbi_disease_bigbio_kb                            6881   \n",
       "nlm_gene_bigbio_kb                               15553   \n",
       "nlmchem_bigbio_kb                                38339   \n",
       "osiris_bigbio_kb                                  1230   \n",
       "pubtator_central_sample_bigbio_kb                  208   \n",
       "quaero_emea_bigbio_kb                             7159   \n",
       "quaero_medline_bigbio_kb                          9074   \n",
       "spl_adr_200db_train_bigbio_kb                    15722   \n",
       "spl_adr_200db_unannotated_bigbio_kb                  0   \n",
       "thomas2011_bigbio_kb                               527   \n",
       "tmvar_v2_bigbio_kb                                 674   \n",
       "tmvar_v3_bigbio_kb                                7837   \n",
       "twadrl_bigbio_kb                                 50740   \n",
       "\n",
       "                                        entities_normalized_count  \\\n",
       "config_name                                                         \n",
       "ask_a_patient_bigbio_kb                                    173240   \n",
       "bc5cdr_bigbio_kb                                            29335   \n",
       "bionlp_st_2019_bb_bigbio_kb                                  3628   \n",
       "biorelex_bigbio_kb                                          10593   \n",
       "cadec_bigbio_kb                                             15932   \n",
       "cantemist_bigbio_kb                                         16032   \n",
       "citation_gia_test_collection_bigbio_kb                       1382   \n",
       "codiesp_X_bigbio_kb                                         18435   \n",
       "distemist_entities_bigbio_kb                                    0   \n",
       "distemist_linking_bigbio_kb                                  5374   \n",
       "gnormplus_bigbio_kb                                          6430   \n",
       "linnaeus_bigbio_kb                                           4259   \n",
       "linnaeus_filtered_bigbio_kb                                  2851   \n",
       "mantra_gsc_de_emea_bigbio_kb                                  425   \n",
       "mantra_gsc_de_medline_bigbio_kb                               309   \n",
       "mantra_gsc_de_patents_bigbio_kb                               348   \n",
       "mantra_gsc_en_emea_bigbio_kb                                  433   \n",
       "mantra_gsc_en_medline_bigbio_kb                              1177   \n",
       "mantra_gsc_en_patents_bigbio_kb                               353   \n",
       "mantra_gsc_es_emea_bigbio_kb                                  432   \n",
       "mantra_gsc_es_medline_bigbio_kb                               324   \n",
       "mantra_gsc_fr_emea_bigbio_kb                                  431   \n",
       "mantra_gsc_fr_medline_bigbio_kb                               279   \n",
       "mantra_gsc_fr_patents_bigbio_kb                               342   \n",
       "mantra_gsc_nl_emea_bigbio_kb                                  434   \n",
       "mantra_gsc_nl_medline_bigbio_kb                               243   \n",
       "medmentions_full_bigbio_kb                                 385098   \n",
       "medmentions_st21pv_bigbio_kb                               203282   \n",
       "mirna_bigbio_kb                                              8318   \n",
       "muchmore_bigbio_kb                                         399189   \n",
       "muchmore_de_bigbio_kb                                      121306   \n",
       "muchmore_en_bigbio_kb                                      277883   \n",
       "ncbi_disease_bigbio_kb                                       7054   \n",
       "nlm_gene_bigbio_kb                                          18056   \n",
       "nlmchem_bigbio_kb                                           41300   \n",
       "osiris_bigbio_kb                                             1230   \n",
       "pubtator_central_sample_bigbio_kb                             208   \n",
       "quaero_emea_bigbio_kb                                        7202   \n",
       "quaero_medline_bigbio_kb                                     9079   \n",
       "spl_adr_200db_train_bigbio_kb                                2417   \n",
       "spl_adr_200db_unannotated_bigbio_kb                             0   \n",
       "thomas2011_bigbio_kb                                          527   \n",
       "tmvar_v2_bigbio_kb                                            625   \n",
       "tmvar_v3_bigbio_kb                                           8884   \n",
       "twadrl_bigbio_kb                                            50740   \n",
       "\n",
       "                                                                    entities_type_counter  \\\n",
       "config_name                                                                                 \n",
       "ask_a_patient_bigbio_kb                                     {'social_media_text': 173240}   \n",
       "bc5cdr_bigbio_kb                                    {'Chemical': 15953, 'Disease': 13318}   \n",
       "bionlp_st_2019_bb_bigbio_kb             {'Habitat': 1752, 'Microorganism': 1174, 'Phen...   \n",
       "biorelex_bigbio_kb                      {'protein': 3730, 'protein-family': 1125, 'che...   \n",
       "cadec_bigbio_kb                         {'ADR': 6318, 'Drug': 1800, 'Finding': 435, 'D...   \n",
       "cantemist_bigbio_kb                                       {'MORFOLOGIA_NEOPLASIA': 16032}   \n",
       "citation_gia_test_collection_bigbio_kb  {'Gene': 1205, 'FamilyName': 160, 'DomainMotif...   \n",
       "codiesp_X_bigbio_kb                         {'DIAGNOSTICO': 14305, 'PROCEDIMIENTO': 4130}   \n",
       "distemist_entities_bigbio_kb                                         {'ENFERMEDAD': 8065}   \n",
       "distemist_linking_bigbio_kb                                          {'ENFERMEDAD': 5136}   \n",
       "gnormplus_bigbio_kb                     {'Gene': 6251, 'FamilyName': 2367, 'DomainMoti...   \n",
       "linnaeus_bigbio_kb                                                      {'species': 4259}   \n",
       "linnaeus_filtered_bigbio_kb                                             {'species': 2851}   \n",
       "mantra_gsc_de_emea_bigbio_kb            {'DISO': 151, 'CHEM': 99, 'PROC': 57, 'LIVB': ...   \n",
       "mantra_gsc_de_medline_bigbio_kb         {'DISO': 77, 'PROC': 71, 'ANAT': 56, 'CHEM': 5...   \n",
       "mantra_gsc_de_patents_bigbio_kb         {'CHEM': 149, 'DISO': 101, 'ANAT': 31, 'PROC':...   \n",
       "mantra_gsc_en_emea_bigbio_kb            {'DISO': 150, 'CHEM': 100, 'PROC': 62, 'LIVB':...   \n",
       "mantra_gsc_en_medline_bigbio_kb         {'DISO': 359, 'PROC': 250, 'LIVB': 157, 'ANAT'...   \n",
       "mantra_gsc_en_patents_bigbio_kb         {'CHEM': 150, 'DISO': 104, 'ANAT': 31, 'PROC':...   \n",
       "mantra_gsc_es_emea_bigbio_kb            {'DISO': 151, 'CHEM': 98, 'PROC': 66, 'LIVB': ...   \n",
       "mantra_gsc_es_medline_bigbio_kb         {'DISO': 115, 'PROC': 63, 'LIVB': 42, 'ANAT': ...   \n",
       "mantra_gsc_fr_emea_bigbio_kb            {'DISO': 151, 'CHEM': 101, 'PROC': 59, 'LIVB':...   \n",
       "mantra_gsc_fr_medline_bigbio_kb         {'DISO': 86, 'PROC': 64, 'LIVB': 45, 'ANAT': 3...   \n",
       "mantra_gsc_fr_patents_bigbio_kb         {'CHEM': 146, 'DISO': 103, 'ANAT': 31, 'PROC':...   \n",
       "mantra_gsc_nl_emea_bigbio_kb            {'DISO': 159, 'CHEM': 101, 'PROC': 57, 'LIVB':...   \n",
       "mantra_gsc_nl_medline_bigbio_kb         {'DISO': 74, 'PROC': 49, 'LIVB': 49, 'ANAT': 2...   \n",
       "medmentions_full_bigbio_kb              {'T080': 31485, 'T169': 23661, 'T081': 19995, ...   \n",
       "medmentions_st21pv_bigbio_kb            {'T038': 41422, 'T103': 37401, 'T058': 24306, ...   \n",
       "mirna_bigbio_kb                         {'Diseases': 2161, 'Relation_Trigger': 1960, '...   \n",
       "muchmore_bigbio_kb                                                   {'umlsterm': 345838}   \n",
       "muchmore_de_bigbio_kb                                                {'umlsterm': 112596}   \n",
       "muchmore_en_bigbio_kb                                                {'umlsterm': 233242}   \n",
       "ncbi_disease_bigbio_kb                  {'SpecificDisease': 3929, 'Modifier': 1766, 'D...   \n",
       "nlm_gene_bigbio_kb                      {'Gene': 9562, 'GENERIF': 3887, 'STARGENE': 19...   \n",
       "nlmchem_bigbio_kb                                                     {'Chemical': 38339}   \n",
       "osiris_bigbio_kb                                            {'gene': 689, 'variant': 541}   \n",
       "pubtator_central_sample_bigbio_kb       {'Disease': 77, 'Chemical': 62, 'Species': 49,...   \n",
       "quaero_emea_bigbio_kb                   {'DISO': 1510, 'CHEM': 2482, 'PROC': 952, 'LIV...   \n",
       "quaero_medline_bigbio_kb                {'DISO': 2843, 'PROC': 1750, 'ANAT': 1499, 'CH...   \n",
       "spl_adr_200db_train_bigbio_kb           {'AdverseReaction': 13795, 'Severity': 934, 'F...   \n",
       "spl_adr_200db_unannotated_bigbio_kb                                                    {}   \n",
       "thomas2011_bigbio_kb                    {'Protein Sequence Mutation': 283, 'Nucleotide...   \n",
       "tmvar_v2_bigbio_kb                      {'ProteinMutation': 436, 'DNAMutation': 190, '...   \n",
       "tmvar_v3_bigbio_kb                      {'['Gene']': 4059, '['Species']': 1756, '['Pro...   \n",
       "twadrl_bigbio_kb                                             {'social_media_text': 50740}   \n",
       "\n",
       "                                                                 entities_db_name_counter  \\\n",
       "config_name                                                                                 \n",
       "ask_a_patient_bigbio_kb                                         {'SNOMED-CT|AMT': 173240}   \n",
       "bc5cdr_bigbio_kb                                                          {'MESH': 29335}   \n",
       "bionlp_st_2019_bb_bigbio_kb                  {'OntoBiotope': 2452, 'NCBI_Taxonomy': 1176}   \n",
       "biorelex_bigbio_kb                      {'uniprot': 3190, 'hgnc': 2947, 'NCBI gene': 2...   \n",
       "cadec_bigbio_kb                              {'Snomed CT': 8930, 'Meddra': 6222, '': 780}   \n",
       "cantemist_bigbio_kb                                                 {'eCIE-O-3.1': 16032}   \n",
       "citation_gia_test_collection_bigbio_kb                                         {'': 1382}   \n",
       "codiesp_X_bigbio_kb                                {'ICD10-CM': 14305, 'ICD10-PCS': 4130}   \n",
       "distemist_entities_bigbio_kb                                                           {}   \n",
       "distemist_linking_bigbio_kb                                           {'SNOMED_CT': 5374}   \n",
       "gnormplus_bigbio_kb                                                    {'NCBIGene': 6430}   \n",
       "linnaeus_bigbio_kb                                                         {'ncbi': 4259}   \n",
       "linnaeus_filtered_bigbio_kb                                                {'ncbi': 2851}   \n",
       "mantra_gsc_de_emea_bigbio_kb                                                {'UMLS': 425}   \n",
       "mantra_gsc_de_medline_bigbio_kb                                             {'UMLS': 309}   \n",
       "mantra_gsc_de_patents_bigbio_kb                                             {'UMLS': 348}   \n",
       "mantra_gsc_en_emea_bigbio_kb                                                {'UMLS': 433}   \n",
       "mantra_gsc_en_medline_bigbio_kb                                            {'UMLS': 1177}   \n",
       "mantra_gsc_en_patents_bigbio_kb                                             {'UMLS': 353}   \n",
       "mantra_gsc_es_emea_bigbio_kb                                                {'UMLS': 432}   \n",
       "mantra_gsc_es_medline_bigbio_kb                                             {'UMLS': 324}   \n",
       "mantra_gsc_fr_emea_bigbio_kb                                                {'UMLS': 431}   \n",
       "mantra_gsc_fr_medline_bigbio_kb                                             {'UMLS': 279}   \n",
       "mantra_gsc_fr_patents_bigbio_kb                                             {'UMLS': 342}   \n",
       "mantra_gsc_nl_emea_bigbio_kb                                                {'UMLS': 434}   \n",
       "mantra_gsc_nl_medline_bigbio_kb                                             {'UMLS': 243}   \n",
       "medmentions_full_bigbio_kb                                               {'UMLS': 385098}   \n",
       "medmentions_st21pv_bigbio_kb                                             {'UMLS': 203282}   \n",
       "mirna_bigbio_kb                                                    {'miRNA-corpus': 8318}   \n",
       "muchmore_bigbio_kb                                                       {'UMLS': 399189}   \n",
       "muchmore_de_bigbio_kb                                                    {'UMLS': 121306}   \n",
       "muchmore_en_bigbio_kb                                                    {'UMLS': 277883}   \n",
       "ncbi_disease_bigbio_kb                                        {'MESH': 6487, 'OMIM': 567}   \n",
       "nlm_gene_bigbio_kb                                                    {'NCBIGene': 18056}   \n",
       "nlmchem_bigbio_kb                                                         {'MESH': 41300}   \n",
       "osiris_bigbio_kb                        {'NCBI Gene': 689, 'HGVS-like': 291, 'dbSNP': ...   \n",
       "pubtator_central_sample_bigbio_kb       {'mesh': 139, 'ncbi_taxon': 49, 'ncbi_gene': 1...   \n",
       "quaero_emea_bigbio_kb                                                      {'UMLS': 7202}   \n",
       "quaero_medline_bigbio_kb                                                   {'UMLS': 9079}   \n",
       "spl_adr_200db_train_bigbio_kb                                      {'MedDRA v18.1': 2417}   \n",
       "spl_adr_200db_unannotated_bigbio_kb                                                    {}   \n",
       "thomas2011_bigbio_kb                                                       {'dbSNP': 527}   \n",
       "tmvar_v2_bigbio_kb                                                         {'dbsnp': 625}   \n",
       "tmvar_v3_bigbio_kb                                     {'NCBI Gene': 6023, 'dbSNP': 2861}   \n",
       "twadrl_bigbio_kb                                                       {'SIDER 4': 50740}   \n",
       "\n",
       "                                        entities_unique_db_ids_count  \n",
       "config_name                                                           \n",
       "ask_a_patient_bigbio_kb                                        16548  \n",
       "bc5cdr_bigbio_kb                                                3907  \n",
       "bionlp_st_2019_bb_bigbio_kb                                      834  \n",
       "biorelex_bigbio_kb                                              3291  \n",
       "cadec_bigbio_kb                                                 1736  \n",
       "cantemist_bigbio_kb                                             1399  \n",
       "citation_gia_test_collection_bigbio_kb                             1  \n",
       "codiesp_X_bigbio_kb                                             5377  \n",
       "distemist_entities_bigbio_kb                                       0  \n",
       "distemist_linking_bigbio_kb                                     2430  \n",
       "gnormplus_bigbio_kb                                             1465  \n",
       "linnaeus_bigbio_kb                                               233  \n",
       "linnaeus_filtered_bigbio_kb                                      233  \n",
       "mantra_gsc_de_emea_bigbio_kb                                     299  \n",
       "mantra_gsc_de_medline_bigbio_kb                                  259  \n",
       "mantra_gsc_de_patents_bigbio_kb                                  223  \n",
       "mantra_gsc_en_emea_bigbio_kb                                     301  \n",
       "mantra_gsc_en_medline_bigbio_kb                                  884  \n",
       "mantra_gsc_en_patents_bigbio_kb                                  227  \n",
       "mantra_gsc_es_emea_bigbio_kb                                     295  \n",
       "mantra_gsc_es_medline_bigbio_kb                                  285  \n",
       "mantra_gsc_fr_emea_bigbio_kb                                     304  \n",
       "mantra_gsc_fr_medline_bigbio_kb                                  242  \n",
       "mantra_gsc_fr_patents_bigbio_kb                                  223  \n",
       "mantra_gsc_nl_emea_bigbio_kb                                     304  \n",
       "mantra_gsc_nl_medline_bigbio_kb                                  204  \n",
       "medmentions_full_bigbio_kb                                     50720  \n",
       "medmentions_st21pv_bigbio_kb                                   35620  \n",
       "mirna_bigbio_kb                                                 8315  \n",
       "muchmore_bigbio_kb                                             10004  \n",
       "muchmore_de_bigbio_kb                                           5573  \n",
       "muchmore_en_bigbio_kb                                           9176  \n",
       "ncbi_disease_bigbio_kb                                          1015  \n",
       "nlm_gene_bigbio_kb                                              3940  \n",
       "nlmchem_bigbio_kb                                               2486  \n",
       "osiris_bigbio_kb                                                 423  \n",
       "pubtator_central_sample_bigbio_kb                                 58  \n",
       "quaero_emea_bigbio_kb                                           1643  \n",
       "quaero_medline_bigbio_kb                                        5616  \n",
       "spl_adr_200db_train_bigbio_kb                                    637  \n",
       "spl_adr_200db_unannotated_bigbio_kb                                0  \n",
       "thomas2011_bigbio_kb                                             385  \n",
       "tmvar_v2_bigbio_kb                                               275  \n",
       "tmvar_v3_bigbio_kb                                              2161  \n",
       "twadrl_bigbio_kb                                               23819  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols_to_keep = ['dataset_name', 'display_name', 'samples_count', 'passages_count', 'passages_char_count',\n",
    "    'passages_type_counter', 'entities_count', 'entities_normalized_count',\n",
    "    'entities_type_counter', 'entities_db_name_counter',\n",
    "    'entities_unique_db_ids_count', 'split',]\n",
    "\n",
    "def dict_agg(x):\n",
    "    output = defaultdict(int)\n",
    "    for d in x:\n",
    "        for k, v in d.items():\n",
    "            output[k] += v\n",
    "    return dict(output)\n",
    "\n",
    "\n",
    "# df.loc[~df.dataset_name.isin(['muchmore','twadrl','ask_a_patient', 'mantra_gsc']), cols_to_keep]\n",
    "\n",
    "aggregated = df.groupby('config_name').agg({\n",
    "    'display_name': 'first',\n",
    "    'dataset_name': 'first',\n",
    "    'samples_count': 'sum', \n",
    "    'passages_count': 'sum', \n",
    "    'passages_char_count': 'sum',\n",
    "    'passages_type_counter': lambda x: dict_agg(x), \n",
    "    'entities_count': 'sum', \n",
    "    'entities_normalized_count': 'sum',\n",
    "    'entities_type_counter': lambda x: dict_agg(x), \n",
    "    'entities_db_name_counter': lambda x: dict_agg(x),\n",
    "    'entities_unique_db_ids_count': 'sum',\n",
    "})\n",
    "\n",
    "# aggregated.loc[~aggregated.dataset_name.isin(['muchmore','twadrl','ask_a_patient', 'mantra_gsc'])].to_excel('../collaboration/dataset_summary.xlsx')\n",
    "# aggregated.loc[~aggregated.dataset_name.isin(['muchmore','twadrl','ask_a_patient', 'mantra_gsc'])]\n",
    "\n",
    "aggregated\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_db_names = defaultdict(list)\n",
    "\n",
    "aggregated[['dataset_name','']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found cached dataset bc5cdr (/home/dkartchner3/.cache/huggingface/datasets/bc5cdr/bc5cdr_bigbio_kb/1.0.0/f01f16ea9b65ead985bedadf7335195c32297c8f1b09417fc607b102a6757d6f)\n"
     ]
    },
    {
     "data": {
      "application/json": {
       "ascii": false,
       "bar_format": null,
       "colour": null,
       "elapsed": 0.012157917022705078,
       "initial": 0,
       "n": 0,
       "ncols": null,
       "nrows": null,
       "postfix": null,
       "prefix": "",
       "rate": null,
       "total": 3,
       "unit": "it",
       "unit_divisor": 1000,
       "unit_scale": false
      },
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2a612c33bd644aa2b91a659979910440",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/3 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata = ner_helpers.for_config_name('bc5cdr_bigbio_kb').get_metadata()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "train = metadata['train']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['train', 'test', 'validation'])"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'samples_count': 500,\n",
       " 'passages_count': 1000,\n",
       " 'passages_char_count': 652177,\n",
       " 'passages_type_counter': {'title': 500, 'abstract': 500},\n",
       " 'entities_count': 9570,\n",
       " 'entities_normalized_count': 9599,\n",
       " 'entities_type_counter': {'Chemical': 5207, 'Disease': 4363},\n",
       " 'entities_db_name_counter': {'MESH': 9599},\n",
       " 'entities_unique_db_ids_count': 1328,\n",
       " 'events_count': 0,\n",
       " 'events_type_counter': {},\n",
       " 'events_arguments_count': 0,\n",
       " 'events_arguments_role_counter': {},\n",
       " 'coreferences_count': 0,\n",
       " 'relations_count': 15072,\n",
       " 'relations_type_counter': {'CID': 15072},\n",
       " 'relations_db_name_counter': {},\n",
       " 'relations_unique_db_ids_count': 0}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.__dict__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel('../collaboration/dataset_summary.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "chemspider\n",
      "dbsnp\n",
      "dbsnp\n",
      "dbSNP\n",
      "DrugBank\n",
      "eCIE-O-3.1\n",
      "hgnc\n",
      "HGVS-like\n",
      "ICD10-CM\n",
      "ICD10-PCS\n",
      "intenz\n",
      "interpro\n",
      "kegg\n",
      "Meddra\n",
      "MedDRA v18.1\n",
      "mesh\n",
      "mesh\n",
      "MESH\n",
      "miRNA-corpus\n",
      "ncbi\n",
      "ncbi gene\n",
      "NCBI Gene\n",
      "NCBI gene\n",
      "ncbi_gene\n",
      "ncbi_taxon\n",
      "NCBI_Taxonomy\n",
      "NCBIGene\n",
      "OMIM\n",
      "OntoBiotope\n",
      "pfam\n",
      "pubchem:compound\n",
      "pubchem:substance\n",
      "Snomed CT\n",
      "SNOMED_CT\n",
      "tmVar\n",
      "UMLS\n",
      "uniprot\n"
     ]
    }
   ],
   "source": [
    "import ast\n",
    "db_names = defaultdict(set)\n",
    "for d in df.entities_db_name_counter.tolist():\n",
    "    db_dict = ast.literal_eval(d)\n",
    "    for x in db_dict.keys():\n",
    "        db_names[x.lower()].add(x)\n",
    "\n",
    "db_keys = list(db_names.keys())\n",
    "db_keys.sort()\n",
    "db_names = {key: list(db_names[key]) for key in db_keys}\n",
    "for key, val in db_names.items():\n",
    "    if len(val) > 1:\n",
    "        print(key)\n",
    "    for v in val:\n",
    "        # if v == key:\n",
    "        #     continue\n",
    "        print(v)\n",
    "\n",
    "with open('../collaboration/db_names.json', 'w') as f:\n",
    "    f.write(ujson.dumps(list(db_names), indent=2))\n",
    "# for x in sorted([x.lower() for x in db_names.keys()]):\n",
    "#     print(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['', 'chemspider', 'dbsnp', 'drugbank', 'ecie-o-3.1', 'hgnc', 'hgvs-like', 'icd10-cm', 'icd10-pcs', 'intenz', 'interpro', 'kegg', 'meddra', 'meddra v18.1', 'mesh', 'mirna-corpus', 'ncbi', 'ncbi gene', 'ncbi_gene', 'ncbi_taxon', 'ncbi_taxonomy', 'ncbigene', 'omim', 'ontobiotope', 'pfam', 'pubchem:compound', 'pubchem:substance', 'snomed ct', 'snomed_ct', 'tmvar', 'umls', 'uniprot']\n"
     ]
    }
   ],
   "source": [
    "with open('../collaboration/db_names.txt', 'w') as f:\n",
    "    f.write(\"\\n\".join())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[\"{'MESH': 29335}\",\n",
       " \"{'OntoBiotope': 2452, 'NCBI_Taxonomy': 1176}\",\n",
       " \"{'uniprot': 3190, 'hgnc': 2947, 'NCBI gene': 2946, 'interpro': 816, 'pubchem:compound': 459, 'intenz': 116, 'pfam': 98, 'DrugBank': 13, 'pubchem:substance': 6, 'kegg': 1, 'chemspider': 1}\",\n",
       " \"{'Snomed CT': 8930, 'Meddra': 6222, '': 780}\",\n",
       " \"{'eCIE-O-3.1': 16032}\",\n",
       " \"{'': 1382}\",\n",
       " \"{'ICD10-CM': 14305, 'ICD10-PCS': 4130}\",\n",
       " '{}',\n",
       " \"{'SNOMED_CT': 5374}\",\n",
       " \"{'NCBIGene': 6430}\",\n",
       " \"{'ncbi': 4259}\",\n",
       " \"{'ncbi': 2851}\",\n",
       " \"{'UMLS': 385098}\",\n",
       " \"{'UMLS': 203282}\",\n",
       " \"{'miRNA-corpus': 8318}\",\n",
       " \"{'MESH': 6487, 'OMIM': 567}\",\n",
       " \"{'NCBIGene': 18056}\",\n",
       " \"{'MESH': 41300}\",\n",
       " \"{'NCBI Gene': 689, 'HGVS-like': 291, 'dbSNP': 250}\",\n",
       " \"{'mesh': 139, 'ncbi_taxon': 49, 'ncbi_gene': 11, 'tmVar': 9}\",\n",
       " \"{'UMLS': 7202}\",\n",
       " \"{'UMLS': 9079}\",\n",
       " \"{'MedDRA v18.1': 2417}\",\n",
       " '{}',\n",
       " \"{'dbSNP': 527}\",\n",
       " \"{'dbsnp': 625}\",\n",
       " \"{'NCBI Gene': 6023, 'dbSNP': 2861}\"]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.entities_db_name_counter.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "name_list = ujson.load(open('../collaboration/db_names.json'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "el-robustness",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "b8f429b760dbf9b012b3ac69bc0c20ad885147b11fe7d141cd50535b2d760548"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
